<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Differential Gene Expression (DGE) Analysis</title>

<script src="site_libs/header-attrs-2.22/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<script src="site_libs/plotly-binding-4.10.2/plotly.js"></script>
<script src="site_libs/typedarray-0.1/typedarray.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="site_libs/plotly-main-2.11.1/plotly-latest.min.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">mecfs-dge-analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="background.html">Background</a>
</li>
<li>
  <a href="analysis.html">Analysis</a>
</li>
<li>
  <a href="initial-analysis.html">Initial Analysis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Differential Gene Expression (DGE)
Analysis</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-07-24
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>mecfs-dge-analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230618code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230618)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230618code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230618)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomuabcgdswortheymecfsdgeanalysistree5252856f5e668e0c4c422f7cdb660ca44fcd1cf8targetblank5252856a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/tree/5252856f5e668e0c4c422f7cdb660ca44fcd1cf8" target="_blank">5252856</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomuabcgdswortheymecfsdgeanalysistree5252856f5e668e0c4c422f7cdb660ca44fcd1cf8targetblank5252856a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/tree/5252856f5e668e0c4c422f7cdb660ca44fcd1cf8" target="_blank">5252856</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    output/batch-correction-limma/

Unstaged changes:
    Modified:   analysis/_site.yml
    Modified:   output/counts_vst.csv
    Modified:   output/counts_vst_limma.csv
    Modified:   output/res_aff_vs_unaff.csv
    Modified:   output/res_aff_vs_unaff_df_genename_05.csv
    Modified:   output/res_aff_vs_unaff_genename.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/analysis.Rmd</code>) and HTML
(<code>docs/analysis.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/uab-cgds-worthey/mecfs-dge-analysis/0b76a235f80e48093d721754e30162f4eef00e7b/docs/analysis.html" target="_blank">0b76a23</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-07-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/f6e60363c15bbb96c5e1ced3651003f19e7802d9/analysis/analysis.Rmd" target="_blank">f6e6036</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-07-12
</td>
<td>
wflow_publish("analysis/*")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/uab-cgds-worthey/mecfs-dge-analysis/fad56f5bbc2d6f73cef2a7b2bbbab1754452970d/docs/analysis.html" target="_blank">fad56f5</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-07-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/41b1c38b9ff4bd81ff7b024e736b2d05d979176b/analysis/analysis.Rmd" target="_blank">41b1c38</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-07-06
</td>
<td>
wflow_publish("analysis/*")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/uab-cgds-worthey/mecfs-dge-analysis/ae8df0ec58683a6f7764e9a20cbc61eab6878a30/docs/analysis.html" target="_blank">ae8df0e</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-07-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/091a5135252bf6ef6770d003470ea74a7c272c1b/analysis/analysis.Rmd" target="_blank">091a513</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-07-06
</td>
<td>
wflow_publish("analysis/*")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/uab-cgds-worthey/mecfs-dge-analysis/53d12da39293a4bfb7d8aa6da6614d9a628b9176/docs/analysis.html" target="_blank">53d12da</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-07-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/bbdd7e3e937266d30d0f0605b6052154f6de1571/analysis/analysis.Rmd" target="_blank">bbdd7e3</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-07-06
</td>
<td>
wflow_publish("analysis/*")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/uab-cgds-worthey/mecfs-dge-analysis/ad9ce58ce4495d2e8eebbe4b2e323cf124094cf2/docs/analysis.html" target="_blank">ad9ce58</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-07-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/b5cecfb72c2b020c9f00fefba23959002600bf27/analysis/analysis.Rmd" target="_blank">b5cecfb</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-07-01
</td>
<td>
wflow_publish("analysis/*")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/uab-cgds-worthey/mecfs-dge-analysis/597034daf6d9bfa65ba200fa716e748f8360d8be/docs/analysis.html" target="_blank">597034d</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/uab-cgds-worthey/mecfs-dge-analysis/08f632068d1a715b373d662057f4d57adbcbe5a0/docs/analysis.html" target="_blank">08f6320</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/8895f16e123edea6113724162bea586a461c0ba0/analysis/analysis.Rmd" target="_blank">8895f16</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-28
</td>
<td>
Add code block for package install.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/uab-cgds-worthey/mecfs-dge-analysis/37ab1642bb81d2294fecd18f6c34674682fe7f11/docs/analysis.html" target="_blank">37ab164</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/300901630f5013adb2b8a0b2906eeaa62e2d20c0/analysis/analysis.Rmd" target="_blank">3009016</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-28
</td>
<td>
Add gprofiler.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/dbea49f9745a2726745430154236a30824498754/analysis/analysis.Rmd" target="_blank">dbea49f</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-27
</td>
<td>
Add analysis and update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/uab-cgds-worthey/mecfs-dge-analysis/dbea49f9745a2726745430154236a30824498754/docs/analysis.html" target="_blank">dbea49f</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-27
</td>
<td>
Add analysis and update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/uab-cgds-worthey/mecfs-dge-analysis/19c181e90936e7e159bfacaaeaf7a602a7940b96/docs/analysis.html" target="_blank">19c181e</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/13c1acc043d29c30392f20ae81dcc31643366350/analysis/analysis.Rmd" target="_blank">13c1acc</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-23
</td>
<td>
wflow_publish("analysis/")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/uab-cgds-worthey/mecfs-dge-analysis/6133d80bd1e4bcfaa02fe0a71a89feec0506e7a2/docs/analysis.html" target="_blank">6133d80</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/df39c694f4c82d2df0924b35abe610451265461c/analysis/analysis.Rmd" target="_blank">df39c69</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-23
</td>
<td>
wflow_publish("analysis/")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="dge-analysis-setup" class="section level1 tabset">
<h1 class="tabset">DGE Analysis Setup</h1>
<p>Ensure you have all necessary libraries installed and load the helper
code.</p>
<p>At a later date, <code>renv</code> will be integrated to ensure
reproducibility of this analysis.</p>
<p>Use the below code to install these packages:</p>
<pre><code># Install packages from CRAN
install.packages(c(&quot;tidyverse&quot;, &quot;RColorBrewer&quot;, &quot;pheatmap&quot;, &quot;gprofiler2&quot;, &quot;plotly&quot;))

# Install packages from Bioconductor
install.packages(&quot;BiocManager&quot;)
BiocManager::install(c(&quot;DESeq2&quot;, &quot;genefilter&quot;, &quot;limma&quot;, &quot;biomaRt&quot;))</code></pre>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse) <span class="co"># Available via CRAN</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">library</span>(DESeq2) <span class="co"># Available via Bioconductor</span></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer) <span class="co"># Available via CRAN</span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="fu">library</span>(pheatmap) <span class="co"># Available via CRAN</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="fu">library</span>(genefilter) <span class="co"># Available via Bioconductor</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="fu">library</span>(limma) <span class="co"># Available via Bioconductor</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="fu">library</span>(gprofiler2) <span class="co"># Available via CRAN</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="fu">library</span>(biomaRt) <span class="co"># Available via Bioconductor</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="fu">library</span>(plotly) <span class="co"># Available via CRAN</span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a><span class="fu">library</span>(rmarkdown)</span></code></pre></div>
<div id="data-import" class="section level2">
<h2>Data Import</h2>
<p>We will be importing counts data from the star-salmon pipeline and
our metadata for the project which is hosted on Box. This also ensures
data is properly ordered by sample id.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;data/star-salmon/salmon.merged.gene_counts_length_scaled.tsv&quot;</span>)</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="co"># Use first column (gene_id) for row names</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>counts <span class="ot">=</span> <span class="fu">data.frame</span>(counts, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>counts<span class="sc">$</span>Ensembl_ID <span class="ot">=</span> <span class="fu">row.names</span>(counts)</span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>drop <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Ensembl_ID&quot;</span>,<span class="st">&quot;gene_name&quot;</span>)</span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>gene_info <span class="ot">=</span> counts[,drop]    </span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>counts <span class="ot">=</span> counts[ , <span class="sc">!</span>(<span class="fu">names</span>(counts) <span class="sc">%in%</span> drop)]     <span class="co"># remove both columns</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a><span class="co"># Import metadata</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a>sample_metadata <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/MECFS_RNAseq_metadata_2023_06_23.csv&quot;</span>)</span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a><span class="fu">row.names</span>(sample_metadata) <span class="ot">&lt;-</span> sample_metadata<span class="sc">$</span>RNA_Samples_id</span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a></span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a><span class="co"># Check that data is ordered properly</span></span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a><span class="fu">check_order</span>(<span class="at">sample_metadata =</span> sample_metadata, <span class="at">counts =</span> counts)</span></code></pre></div>
<pre><code>[1] &quot;Data matches and is ordered by sample id.&quot;</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>genes_biomart <span class="ot">&lt;-</span> <span class="fu">retrieve_gene_info</span>(<span class="at">values =</span> gene_info<span class="sc">$</span>Ensembl_ID, <span class="at">filters =</span> <span class="st">&#39;ensembl_gene_id_version&#39;</span>)</span></code></pre></div>
</div>
<div id="deseq2-analysis" class="section level2">
<h2>DESeq2 Analysis</h2>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>sample_metadata<span class="sc">$</span>Family <span class="ot">=</span> <span class="fu">factor</span>(sample_metadata<span class="sc">$</span>Family)  </span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>sample_metadata<span class="sc">$</span>Affected <span class="ot">=</span> <span class="fu">factor</span>(sample_metadata<span class="sc">$</span>Affected) </span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>sample_metadata<span class="sc">$</span>Batch <span class="ot">=</span> <span class="fu">factor</span>(sample_metadata<span class="sc">$</span>Batch)</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>sample_metadata<span class="sc">$</span>Gender <span class="ot">=</span> <span class="fu">factor</span>(sample_metadata<span class="sc">$</span>Gender)</span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a><span class="co"># Account for Family later but batch is accounted for</span></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> <span class="fu">round</span>(counts), <span class="at">colData =</span> sample_metadata, <span class="at">design =</span> <span class="sc">~</span> Batch <span class="sc">+</span> Affected)</span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a></span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a><span class="co"># Pre-filtering: Keep only rows that have at least 10 reads total</span></span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a>keep <span class="ot">=</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds)) <span class="sc">&gt;=</span> <span class="dv">10</span> </span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a>dds <span class="ot">=</span> dds[keep,]</span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a></span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a><span class="co"># Run DESeq function</span></span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a>dds <span class="ot">=</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb6-15"><a href="#cb6-15" tabindex="-1"></a></span>
<span id="cb6-16"><a href="#cb6-16" tabindex="-1"></a><span class="co"># Normalize gene counts for differences in seq. depth/global differences</span></span>
<span id="cb6-17"><a href="#cb6-17" tabindex="-1"></a>counts_norm <span class="ot">=</span> <span class="fu">counts</span>(dds, <span class="at">normalized=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<div id="data-transformation-and-visualization" class="section level3">
<h3>Data transformation and visualization</h3>
<p>Perform count data transformation by variance stabilizing
transformation (vst) on normalized counts.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>vsd <span class="ot">=</span> <span class="fu">vst</span>(dds, <span class="at">blind=</span><span class="cn">FALSE</span>) </span></code></pre></div>
</div>
<div id="batch-correction-with-limma" class="section level3">
<h3>Batch correction with limma</h3>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>counts_vst <span class="ot">=</span> <span class="fu">assay</span>(vsd)</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="fu">write.csv</span>(counts_vst, <span class="at">file=</span><span class="st">&quot;output/counts_vst.csv&quot;</span>)</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>mm <span class="ot">=</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> Family <span class="sc">+</span> Affected, <span class="fu">colData</span>(vsd))</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>counts_vst_limma <span class="ot">=</span> limma<span class="sc">::</span><span class="fu">removeBatchEffect</span>(counts_vst, <span class="at">batch=</span>vsd<span class="sc">$</span>Batch, <span class="at">design=</span>mm)</span></code></pre></div>
<pre><code>Coefficients not estimable: batch2 </code></pre>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="fu">write.csv</span>(counts_vst_limma, <span class="at">file=</span><span class="st">&quot;output/counts_vst_limma.csv&quot;</span>)</span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>vsd_limma <span class="ot">=</span> vsd</span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a><span class="fu">assay</span>(vsd_limma) <span class="ot">=</span> counts_vst_limma</span></code></pre></div>
</div>
<div id="sample-distances-heatmap" class="section level3">
<h3>Sample distances heatmap</h3>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>sampleDists <span class="ot">=</span> <span class="fu">dist</span>(<span class="fu">t</span>(<span class="fu">assay</span>(vsd_limma)))</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>sampleDistMatrix <span class="ot">=</span> <span class="fu">as.matrix</span>(sampleDists) </span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a><span class="fu">rownames</span>(sampleDistMatrix) <span class="ot">=</span> <span class="fu">paste</span>(vsd_limma<span class="sc">$</span>Batch, vsd_limma<span class="sc">$</span>Family, <span class="at">sep=</span><span class="st">&quot; | &quot;</span>) </span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="fu">colnames</span>(sampleDistMatrix) <span class="ot">=</span> <span class="fu">paste</span>(vsd_limma<span class="sc">$</span>RNA_Samples_id, vsd_limma<span class="sc">$</span>Family, <span class="at">sep=</span><span class="st">&quot; | &quot;</span>) </span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a>colors <span class="ot">=</span> <span class="fu">colorRampPalette</span>(<span class="fu">rev</span>(<span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">&quot;Blues&quot;</span>)))(<span class="dv">255</span>) </span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a></span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a><span class="fu">pheatmap</span>(sampleDistMatrix, <span class="at">clustering_distance_rows=</span>sampleDists, <span class="at">clustering_distance_cols=</span>sampleDists, <span class="at">col=</span>colors)</span></code></pre></div>
<p><img src="figure/analysis.Rmd/sample-dist-heatmap-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-sample-dist-heatmap-1">
Past versions of sample-dist-heatmap-1.png
</button>
</p>
<div id="fig-sample-dist-heatmap-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/ad9ce58ce4495d2e8eebbe4b2e323cf124094cf2/docs/figure/analysis.Rmd/sample-dist-heatmap-1.png" target="_blank">ad9ce58</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-07-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/08f632068d1a715b373d662057f4d57adbcbe5a0/docs/figure/analysis.Rmd/sample-dist-heatmap-1.png" target="_blank">08f6320</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="principal-components-analysis" class="section level3">
<h3>Principal Components Analysis</h3>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a>pcaData <span class="ot">=</span> <span class="fu">plotPCA</span>(vsd, <span class="at">intgroup=</span><span class="fu">c</span>(<span class="st">&quot;Batch&quot;</span>, <span class="st">&quot;Family&quot;</span>, <span class="st">&quot;Affected&quot;</span>), <span class="at">returnData=</span><span class="cn">TRUE</span>) </span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a>percentVar <span class="ot">=</span> <span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> <span class="fu">attr</span>(pcaData, <span class="st">&quot;percentVar&quot;</span>)) </span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a></span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pcaData, <span class="fu">aes</span>(PC1, PC2, <span class="at">shape=</span><span class="fu">factor</span>(Batch), <span class="at">fill=</span><span class="fu">factor</span>(Affected),  <span class="at">color=</span><span class="fu">factor</span>(Family))) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">5</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC1: &quot;</span>,percentVar[<span class="dv">1</span>],<span class="st">&quot;% variance&quot;</span>)) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC2: &quot;</span>,percentVar[<span class="dv">2</span>],<span class="st">&quot;% variance&quot;</span>)) <span class="sc">+</span> <span class="fu">coord_fixed</span>()</span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a>p1</span></code></pre></div>
<p><img src="figure/analysis.Rmd/principal-components-analysis-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-principal-components-analysis-1">
Past versions of principal-components-analysis-1.png
</button>
</p>
<div id="fig-principal-components-analysis-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/ad9ce58ce4495d2e8eebbe4b2e323cf124094cf2/docs/figure/analysis.Rmd/principal-components-analysis-1.png" target="_blank">ad9ce58</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-07-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/08f632068d1a715b373d662057f4d57adbcbe5a0/docs/figure/analysis.Rmd/principal-components-analysis-1.png" target="_blank">08f6320</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="fu">ggplotly</span>(p1)</span></code></pre></div>
<div class="plotly html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-e56eaee8d12915f7b927" style="width:672px;height:480px;"></div>
<script type="application/json" data-for="htmlwidget-e56eaee8d12915f7b927">{"x":{"data":[{"x":[2.2925865798256257],"y":[5.2203950149491716],"text":"PC1:   2.2925866<br />PC2:   5.2203950<br />factor(Batch): B1<br />factor(Affected): No<br />factor(Family): 1","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(248,118,109,1)","opacity":1,"size":18.897637795275593,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(248,118,109,1)"}},"hoveron":"points","name":"(B1,No,1)","legendgroup":"(B1,No,1)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-3.7531467269071155],"y":[-21.292360627731632],"text":"PC1:  -3.7531467<br />PC2: -21.2923606<br />factor(Batch): B1<br />factor(Affected): No<br />factor(Family): 2","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(248,118,109,1)","opacity":1,"size":18.897637795275593,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(230,134,19,1)"}},"hoveron":"points","name":"(B1,No,2)","legendgroup":"(B1,No,2)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[7.7762026606073658],"y":[2.701663012025322],"text":"PC1:   7.7762027<br />PC2:   2.7016630<br />factor(Batch): B1<br />factor(Affected): No<br />factor(Family): 3","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(248,118,109,1)","opacity":1,"size":18.897637795275593,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(205,150,0,1)"}},"hoveron":"points","name":"(B1,No,3)","legendgroup":"(B1,No,3)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-6.088447398591522],"y":[7.467526359359792],"text":"PC1:  -6.0884474<br />PC2:   7.4675264<br />factor(Batch): B1<br />factor(Affected): No<br />factor(Family): 4","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(248,118,109,1)","opacity":1,"size":18.897637795275593,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(171,163,0,1)"}},"hoveron":"points","name":"(B1,No,4)","legendgroup":"(B1,No,4)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[12.752953252848876],"y":[-25.166040687834723],"text":"PC1:  12.7529533<br />PC2: -25.1660407<br />factor(Batch): B1<br />factor(Affected): No<br />factor(Family): 5","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(248,118,109,1)","opacity":1,"size":18.897637795275593,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(124,174,0,1)"}},"hoveron":"points","name":"(B1,No,5)","legendgroup":"(B1,No,5)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[8.9943355492494117],"y":[2.6644579757817577],"text":"PC1:   8.9943355<br />PC2:   2.6644580<br />factor(Batch): B1<br />factor(Affected): No<br />factor(Family): 6","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(248,118,109,1)","opacity":1,"size":18.897637795275593,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(12,183,2,1)"}},"hoveron":"points","name":"(B1,No,6)","legendgroup":"(B1,No,6)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-6.5820538861605868],"y":[4.6168286200627318],"text":"PC1:  -6.5820539<br />PC2:   4.6168286<br />factor(Batch): B1<br />factor(Affected): Yes<br />factor(Family): 1","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,191,196,1)","opacity":1,"size":18.897637795275593,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(248,118,109,1)"}},"hoveron":"points","name":"(B1,Yes,1)","legendgroup":"(B1,Yes,1)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-20.204210484852315],"y":[3.1226899009900375],"text":"PC1: -20.2042105<br />PC2:   3.1226899<br />factor(Batch): B1<br />factor(Affected): Yes<br />factor(Family): 2","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,191,196,1)","opacity":1,"size":18.897637795275593,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(230,134,19,1)"}},"hoveron":"points","name":"(B1,Yes,2)","legendgroup":"(B1,Yes,2)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[15.023478485103121],"y":[0.99121588367717939],"text":"PC1:  15.0234785<br />PC2:   0.9912159<br />factor(Batch): B1<br />factor(Affected): Yes<br />factor(Family): 3","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,191,196,1)","opacity":1,"size":18.897637795275593,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(205,150,0,1)"}},"hoveron":"points","name":"(B1,Yes,3)","legendgroup":"(B1,Yes,3)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-16.425406958591413],"y":[2.2789544930379284],"text":"PC1: -16.4254070<br />PC2:   2.2789545<br />factor(Batch): B1<br />factor(Affected): Yes<br />factor(Family): 4","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,191,196,1)","opacity":1,"size":18.897637795275593,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(171,163,0,1)"}},"hoveron":"points","name":"(B1,Yes,4)","legendgroup":"(B1,Yes,4)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-0.98122936967863728],"y":[7.6222393441890661],"text":"PC1:  -0.9812294<br />PC2:   7.6222393<br />factor(Batch): B1<br />factor(Affected): Yes<br />factor(Family): 5","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,191,196,1)","opacity":1,"size":18.897637795275593,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(124,174,0,1)"}},"hoveron":"points","name":"(B1,Yes,5)","legendgroup":"(B1,Yes,5)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-5.1361502114851048],"y":[-4.0135084572199364],"text":"PC1:  -5.1361502<br />PC2:  -4.0135085<br />factor(Batch): B1<br />factor(Affected): Yes<br />factor(Family): 6","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,191,196,1)","opacity":1,"size":18.897637795275593,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(12,183,2,1)"}},"hoveron":"points","name":"(B1,Yes,6)","legendgroup":"(B1,Yes,6)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[9.5454162895791512],"y":[-2.9181372097167118],"text":"PC1:   9.5454163<br />PC2:  -2.9181372<br />factor(Batch): B2<br />factor(Affected): No<br />factor(Family): 11","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(248,118,109,1)","opacity":1,"size":18.897637795275593,"symbol":"triangle-up","line":{"width":1.8897637795275593,"color":"rgba(0,169,255,1)"}},"hoveron":"points","name":"(B2,No,11)","legendgroup":"(B2,No,11)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-0.53888579816118587],"y":[-2.0611874093157971],"text":"PC1:  -0.5388858<br />PC2:  -2.0611874<br />factor(Batch): B2<br />factor(Affected): Yes<br />factor(Family): 5","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,191,196,1)","opacity":1,"size":18.897637795275593,"symbol":"triangle-up","line":{"width":1.8897637795275593,"color":"rgba(124,174,0,1)"}},"hoveron":"points","name":"(B2,Yes,5)","legendgroup":"(B2,Yes,5)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[12.468333566102908],"y":[0.7797366055525935],"text":"PC1:  12.4683336<br />PC2:   0.7797366<br />factor(Batch): B2<br />factor(Affected): Yes<br />factor(Family): 6","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,191,196,1)","opacity":1,"size":18.897637795275593,"symbol":"triangle-up","line":{"width":1.8897637795275593,"color":"rgba(12,183,2,1)"}},"hoveron":"points","name":"(B2,Yes,6)","legendgroup":"(B2,Yes,6)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[13.680304555405224],"y":[1.2124118744172954],"text":"PC1:  13.6803046<br />PC2:   1.2124119<br />factor(Batch): B2<br />factor(Affected): Yes<br />factor(Family): 7","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,191,196,1)","opacity":1,"size":18.897637795275593,"symbol":"triangle-up","line":{"width":1.8897637795275593,"color":"rgba(0,190,103,1)"}},"hoveron":"points","name":"(B2,Yes,7)","legendgroup":"(B2,Yes,7)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-9.880202501190265],"y":[-20.529457930319371],"text":"PC1:  -9.8802025<br />PC2: -20.5294579<br />factor(Batch): B2<br />factor(Affected): Yes<br />factor(Family): 8","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,191,196,1)","opacity":1,"size":18.897637795275593,"symbol":"triangle-up","line":{"width":1.8897637795275593,"color":"rgba(0,193,154,1)"}},"hoveron":"points","name":"(B2,Yes,8)","legendgroup":"(B2,Yes,8)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-13.637273253277495,-40.182317570792456],"y":[0.7252047828497058,1.4283013163380169],"text":["PC1: -13.6372733<br />PC2:   0.7252048<br />factor(Batch): B2<br />factor(Affected): Yes<br />factor(Family): 9","PC1: -40.1823176<br />PC2:   1.4283013<br />factor(Batch): B2<br />factor(Affected): Yes<br />factor(Family): 9"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,191,196,1)","opacity":1,"size":18.897637795275593,"symbol":"triangle-up","line":{"width":1.8897637795275593,"color":"rgba(0,191,196,1)"}},"hoveron":"points","name":"(B2,Yes,9)","legendgroup":"(B2,Yes,9)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-16.992549802639015,-4.6536826999052785],"y":[-1.2597795116716706,4.0146523308900086],"text":["PC1: -16.9925498<br />PC2:  -1.2597795<br />factor(Batch): B2<br />factor(Affected): Yes<br />factor(Family): 10","PC1:  -4.6536827<br />PC2:   4.0146523<br />factor(Batch): B2<br />factor(Affected): Yes<br />factor(Family): 10"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,191,196,1)","opacity":1,"size":18.897637795275593,"symbol":"triangle-up","line":{"width":1.8897637795275593,"color":"rgba(0,184,231,1)"}},"hoveron":"points","name":"(B2,Yes,10)","legendgroup":"(B2,Yes,10)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[13.311363454342089],"y":[4.9090948133613503],"text":"PC1:  13.3113635<br />PC2:   4.9090948<br />factor(Batch): B2<br />factor(Affected): Yes<br />factor(Family): 11","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,191,196,1)","opacity":1,"size":18.897637795275593,"symbol":"triangle-up","line":{"width":1.8897637795275593,"color":"rgba(0,169,255,1)"}},"hoveron":"points","name":"(B2,Yes,11)","legendgroup":"(B2,Yes,11)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-6.8050736843946922],"y":[3.3602446001409678],"text":"PC1:  -6.8050737<br />PC2:   3.3602446<br />factor(Batch): B3<br />factor(Affected): Yes<br />factor(Family): 12","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,191,196,1)","opacity":1,"size":18.897637795275593,"symbol":"square","line":{"width":1.8897637795275593,"color":"rgba(132,148,255,1)"}},"hoveron":"points","name":"(B3,Yes,12)","legendgroup":"(B3,Yes,12)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-7.6855063272647586,21.346998779302751,11.757000805333947],"y":[7.0625986365202671,-0.95190488008617624,2.1178205429950019],"text":["PC1:  -7.6855063<br />PC2:   7.0625986<br />factor(Batch): B3<br />factor(Affected): Yes<br />factor(Family): 13","PC1:  21.3469988<br />PC2:  -0.9519049<br />factor(Batch): B3<br />factor(Affected): Yes<br />factor(Family): 13","PC1:  11.7570008<br />PC2:   2.1178205<br />factor(Batch): B3<br />factor(Affected): Yes<br />factor(Family): 13"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,191,196,1)","opacity":1,"size":18.897637795275593,"symbol":"square","line":{"width":1.8897637795275593,"color":"rgba(199,124,255,1)"}},"hoveron":"points","name":"(B3,Yes,13)","legendgroup":"(B3,Yes,13)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-0.29650945283291363],"y":[6.1670755942003943],"text":"PC1:  -0.2965095<br />PC2:   6.1670756<br />factor(Batch): B3<br />factor(Affected): Yes<br />factor(Family): 14","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,191,196,1)","opacity":1,"size":18.897637795275593,"symbol":"square","line":{"width":1.8897637795275593,"color":"rgba(237,104,237,1)"}},"hoveron":"points","name":"(B3,Yes,14)","legendgroup":"(B3,Yes,14)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[26.826187430577711],"y":[7.2092219529085506],"text":"PC1:  26.8261874<br />PC2:   7.2092220<br />factor(Batch): B3<br />factor(Affected): Yes<br />factor(Family): 15","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,191,196,1)","opacity":1,"size":18.897637795275593,"symbol":"square","line":{"width":1.8897637795275593,"color":"rgba(255,97,204,1)"}},"hoveron":"points","name":"(B3,Yes,15)","legendgroup":"(B3,Yes,15)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[4.0674847184465523],"y":[2.5200430596488941],"text":"PC1:   4.0674847<br />PC2:   2.5200431<br />factor(Batch): B3<br />factor(Affected): Yes<br />factor(Family): 16","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,191,196,1)","opacity":1,"size":18.897637795275593,"symbol":"square","line":{"width":1.8897637795275593,"color":"rgba(255,104,161,1)"}},"hoveron":"points","name":"(B3,Yes,16)","legendgroup":"(B3,Yes,16)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":26.228310502283108,"r":7.3059360730593621,"b":40.182648401826498,"l":43.105022831050235},"plot_bgcolor":"rgba(235,235,235,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-43.532742820860967,30.176612680646219],"tickmode":"array","ticktext":["-40","-20","0","20"],"tickvals":[-40,-20,-7.1054273576010019e-15,20],"categoryorder":"array","categoryarray":["-40","-20","0","20"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(255,255,255,1)","gridwidth":0.66417600664176002,"zeroline":false,"anchor":"y","title":{"text":"PC1: 31% variance","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"scaleanchor":"y","scaleratio":1,"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-26.805454689435912,9.2616533457902559],"tickmode":"array","ticktext":["-20","-10","0"],"tickvals":[-20,-10,0],"categoryorder":"array","categoryarray":["-20","-10","0"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(255,255,255,1)","gridwidth":0.66417600664176002,"zeroline":false,"anchor":"x","title":{"text":"PC2: 11% variance","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"scaleanchor":"x","scaleratio":1,"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":true,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.8897637795275593,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498},"title":{"text":"factor(Family)<br />factor(Affected)<br />factor(Batch)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"8f061085e8d5":{"x":{},"y":{},"shape":{},"fill":{},"colour":{},"type":"scatter"}},"cur_data":"8f061085e8d5","visdat":{"8f061085e8d5":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="fu">ggplot</span>(pcaData, <span class="fu">aes</span>(PC1, PC2, <span class="at">shape=</span><span class="fu">factor</span>(Batch), <span class="at">color=</span><span class="fu">factor</span>(Affected)))  <span class="sc">+</span> </span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">5</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC1: &quot;</span>,percentVar[<span class="dv">1</span>],<span class="st">&quot;% variance&quot;</span>)) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC2: &quot;</span>,percentVar[<span class="dv">2</span>],<span class="st">&quot;% variance&quot;</span>)) <span class="sc">+</span> <span class="fu">coord_fixed</span>()</span></code></pre></div>
<p><img src="figure/analysis.Rmd/principal-components-analysis-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-principal-components-analysis-3">
Past versions of principal-components-analysis-3.png
</button>
</p>
<div id="fig-principal-components-analysis-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/ad9ce58ce4495d2e8eebbe4b2e323cf124094cf2/docs/figure/analysis.Rmd/principal-components-analysis-3.png" target="_blank">ad9ce58</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-07-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/08f632068d1a715b373d662057f4d57adbcbe5a0/docs/figure/analysis.Rmd/principal-components-analysis-3.png" target="_blank">08f6320</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="heatmap-of-top-50-top-100-genes" class="section level3">
<h3>Heatmap of top 50 &amp; top 100 genes</h3>
<p>This is a heatmap for 50 genes with the highest variance across
samples.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a>topVarGenes <span class="ot">=</span> <span class="fu">head</span>(<span class="fu">order</span>(<span class="sc">-</span><span class="fu">rowVars</span>(<span class="fu">assay</span>(vsd))),<span class="dv">50</span>)</span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a>mat <span class="ot">=</span> <span class="fu">assay</span>(vsd_limma)[ topVarGenes, ]</span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a>mat <span class="ot">=</span> mat <span class="sc">-</span> <span class="fu">rowMeans</span>(mat)</span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a>df <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(vsd)[,<span class="fu">c</span>(<span class="st">&quot;Batch&quot;</span>, <span class="st">&quot;Affected&quot;</span>)])</span>
<span id="cb15-5"><a href="#cb15-5" tabindex="-1"></a><span class="fu">pheatmap</span>(mat, <span class="at">annotation_col=</span>df, <span class="at">fontsize =</span> <span class="dv">5</span>)</span></code></pre></div>
<p><img src="figure/analysis.Rmd/top50-heatmap-1.png" width="800px" height="570px" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-top50-heatmap-1">
Past versions of top50-heatmap-1.png
</button>
</p>
<div id="fig-top50-heatmap-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/ad9ce58ce4495d2e8eebbe4b2e323cf124094cf2/docs/figure/analysis.Rmd/top50-heatmap-1.png" target="_blank">ad9ce58</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-07-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/dbea49f9745a2726745430154236a30824498754/docs/figure/analysis.Rmd/top50-heatmap-1.png" target="_blank">dbea49f</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>This is a heatmap of the top 100 genes with the highest variance
across samples.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a>topVarGenes <span class="ot">=</span> <span class="fu">head</span>(<span class="fu">order</span>(<span class="sc">-</span><span class="fu">rowVars</span>(<span class="fu">assay</span>(vsd_limma))),<span class="dv">100</span>)</span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a>mat <span class="ot">=</span> <span class="fu">assay</span>(vsd_limma)[ topVarGenes, ]</span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a>mat <span class="ot">=</span> mat <span class="sc">-</span> <span class="fu">rowMeans</span>(mat)</span>
<span id="cb16-4"><a href="#cb16-4" tabindex="-1"></a>df <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(vsd_limma)[,<span class="fu">c</span>(<span class="st">&quot;Batch&quot;</span>, <span class="st">&quot;Family&quot;</span>, <span class="st">&quot;Affected&quot;</span>)])</span>
<span id="cb16-5"><a href="#cb16-5" tabindex="-1"></a><span class="fu">pheatmap</span>(mat, <span class="at">annotation_col=</span>df, <span class="at">fontsize =</span> <span class="dv">6</span>)</span></code></pre></div>
<p><img src="figure/analysis.Rmd/top100-heatmap-1.png" width="1200px" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-top100-heatmap-1">
Past versions of top100-heatmap-1.png
</button>
</p>
<div id="fig-top100-heatmap-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/ad9ce58ce4495d2e8eebbe4b2e323cf124094cf2/docs/figure/analysis.Rmd/top100-heatmap-1.png" target="_blank">ad9ce58</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-07-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/dbea49f9745a2726745430154236a30824498754/docs/figure/analysis.Rmd/top100-heatmap-1.png" target="_blank">dbea49f</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="comparisoncontrast-of-affected_yes_vs_no"
class="section level3">
<h3>Comparison/Contrast of Affected_Yes_vs_No</h3>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a>res_aff_vs_unaff <span class="ot">=</span> <span class="fu">results</span>(dds, <span class="at">contrast=</span><span class="fu">c</span>(<span class="st">&quot;Affected&quot;</span>, <span class="st">&quot;Yes&quot;</span>, <span class="st">&quot;No&quot;</span>))</span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a>res_aff_vs_unaff<span class="ot">=</span> res_aff_vs_unaff[<span class="fu">order</span>(res_aff_vs_unaff<span class="sc">$</span>padj),]</span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a><span class="fu">summary</span>(res_aff_vs_unaff)</span></code></pre></div>
<pre><code>
out of 29623 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 20, 0.068%
LFC &lt; 0 (down)     : 10, 0.034%
outliers [1]       : 134, 0.45%
low counts [2]     : 3997, 13%
(mean count &lt; 1)
[1] see &#39;cooksCutoff&#39; argument of ?results
[2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a><span class="fu">write.csv</span>(res_aff_vs_unaff, <span class="at">file=</span><span class="st">&quot;output/res_aff_vs_unaff.csv&quot;</span>)      </span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a>res_aff_vs_unaff_df <span class="ot">=</span> <span class="fu">as.data.frame</span>(res_aff_vs_unaff)</span>
<span id="cb19-3"><a href="#cb19-3" tabindex="-1"></a>res_aff_vs_unaff_05 <span class="ot">=</span> <span class="fu">subset</span>(res_aff_vs_unaff_df, padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)                             </span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a>topgenes_byensemblid <span class="ot">=</span> <span class="fu">head</span>(<span class="fu">rownames</span>(res_aff_vs_unaff_05),<span class="dv">50</span>)</span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a>topgenes_aff_vs_unaff_05 <span class="ot">=</span> <span class="fu">assay</span>(vsd_limma)[topgenes_byensemblid,] </span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a>topgenes_aff_vs_unaff_05 <span class="ot">=</span> topgenes_aff_vs_unaff_05 <span class="sc">-</span> <span class="fu">rowMeans</span>(topgenes_aff_vs_unaff_05)</span>
<span id="cb20-4"><a href="#cb20-4" tabindex="-1"></a></span>
<span id="cb20-5"><a href="#cb20-5" tabindex="-1"></a><span class="co"># Convert ensemblids</span></span>
<span id="cb20-6"><a href="#cb20-6" tabindex="-1"></a>ensemblids <span class="ot">&lt;-</span> topgenes_byensemblid</span>
<span id="cb20-7"><a href="#cb20-7" tabindex="-1"></a><span class="fu">rownames</span>(topgenes_aff_vs_unaff_05) <span class="ot">&lt;-</span> gene_info<span class="sc">$</span>gene_name[<span class="fu">match</span>(ensemblids, gene_info<span class="sc">$</span>Ensembl_ID)]</span>
<span id="cb20-8"><a href="#cb20-8" tabindex="-1"></a></span>
<span id="cb20-9"><a href="#cb20-9" tabindex="-1"></a>topgenes_aff_vs_unaff_05 <span class="ot">&lt;-</span> topgenes_aff_vs_unaff_05[<span class="fu">order</span>(<span class="fu">row.names</span>(topgenes_aff_vs_unaff_05)), ]</span>
<span id="cb20-10"><a href="#cb20-10" tabindex="-1"></a></span>
<span id="cb20-11"><a href="#cb20-11" tabindex="-1"></a>df <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(vsd_limma)[,<span class="fu">c</span>(<span class="st">&quot;Batch&quot;</span>, <span class="st">&quot;Family&quot;</span>, <span class="st">&quot;Affected&quot;</span>)])</span>
<span id="cb20-12"><a href="#cb20-12" tabindex="-1"></a><span class="fu">pheatmap</span>(topgenes_aff_vs_unaff_05, <span class="at">annotation_col=</span>df, <span class="at">fontsize =</span> <span class="dv">5</span>)</span></code></pre></div>
<p><img src="figure/analysis.Rmd/top%2050%20genes-1.png" width="800px" height="570px" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-top-50-genes-1">
Past versions of "top 50 genes-1.png"
</button>
</p>
<div id="fig-no-spaces-top-50-genes-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/ad9ce58ce4495d2e8eebbe4b2e323cf124094cf2/docs/figure/analysis.Rmd/top 50 genes-1.png" target="_blank">ad9ce58</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-07-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/dbea49f9745a2726745430154236a30824498754/docs/figure/analysis.Rmd/top 50 genes-1.png" target="_blank">dbea49f</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a>res_aff_vs_unaff_df_genename <span class="ot">=</span> res_aff_vs_unaff_df</span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a>res_aff_vs_unaff_df_genename<span class="sc">$</span>Ensembl_ID <span class="ot">=</span> <span class="fu">row.names</span>(res_aff_vs_unaff_df)</span>
<span id="cb21-3"><a href="#cb21-3" tabindex="-1"></a>res_aff_vs_unaff_df_genename <span class="ot">=</span> <span class="fu">merge</span>(<span class="at">x=</span>res_aff_vs_unaff_df_genename, <span class="at">y=</span>gene_info, <span class="at">by.x =</span><span class="st">&quot;Ensembl_ID&quot;</span>, <span class="at">by.y=</span><span class="st">&quot;Ensembl_ID&quot;</span>, <span class="at">all.x=</span>T)</span>
<span id="cb21-4"><a href="#cb21-4" tabindex="-1"></a>res_aff_vs_unaff_df_genename <span class="ot">=</span> res_aff_vs_unaff_df_genename[,<span class="fu">c</span>(<span class="fu">dim</span>(res_aff_vs_unaff_df_genename)[<span class="dv">2</span>],<span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(res_aff_vs_unaff_df_genename)[<span class="dv">2</span>]<span class="sc">-</span><span class="dv">1</span>)]</span>
<span id="cb21-5"><a href="#cb21-5" tabindex="-1"></a>res_aff_vs_unaff_df_genename <span class="ot">=</span> res_aff_vs_unaff_df_genename[<span class="fu">order</span>(res_aff_vs_unaff_df_genename[,<span class="st">&quot;padj&quot;</span>]),]</span>
<span id="cb21-6"><a href="#cb21-6" tabindex="-1"></a><span class="fu">write.csv</span>(res_aff_vs_unaff_df_genename,<span class="at">file=</span><span class="st">&quot;output/res_aff_vs_unaff_genename.csv&quot;</span> )</span></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a>res_aff_vs_unaff_df_genename_05<span class="ot">=</span> <span class="fu">subset</span>(res_aff_vs_unaff_df_genename, padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)                   </span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a>res_aff_vs_unaff_df_genename_05 <span class="ot">=</span> res_aff_vs_unaff_df_genename_05[<span class="fu">order</span>(res_aff_vs_unaff_df_genename_05<span class="sc">$</span>padj),]</span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a><span class="fu">write.csv</span>(res_aff_vs_unaff_df_genename_05, <span class="at">file=</span><span class="st">&quot;output/res_aff_vs_unaff_df_genename_05.csv&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a><span class="co"># Select specific genes to show</span></span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a><span class="co"># set top = 0, then specify genes using label.select argument</span></span>
<span id="cb23-3"><a href="#cb23-3" tabindex="-1"></a><span class="fu">ggmaplot</span>(res_aff_vs_unaff_df_genename, <span class="at">main =</span> <span class="st">&quot;Affected vs Unaffected&quot;</span>,</span>
<span id="cb23-4"><a href="#cb23-4" tabindex="-1"></a>         <span class="at">fdr =</span> .<span class="dv">05</span>, <span class="at">fc =</span> <span class="dv">2</span>, <span class="at">size =</span> <span class="fl">0.4</span>,</span>
<span id="cb23-5"><a href="#cb23-5" tabindex="-1"></a>         <span class="at">genenames =</span> <span class="fu">as.vector</span>(res_aff_vs_unaff_df_genename<span class="sc">$</span>gene_name),</span>
<span id="cb23-6"><a href="#cb23-6" tabindex="-1"></a>         <span class="at">ggtheme =</span> ggplot2<span class="sc">::</span><span class="fu">theme_minimal</span>(),</span>
<span id="cb23-7"><a href="#cb23-7" tabindex="-1"></a>         <span class="at">legend =</span> <span class="st">&quot;top&quot;</span>, <span class="at">top =</span> <span class="dv">19</span>, <span class="at">font.label =</span> <span class="fu">c</span>(<span class="st">&quot;bold&quot;</span>, <span class="dv">8</span>), <span class="at">label.rectangle =</span> <span class="cn">TRUE</span>,</span>
<span id="cb23-8"><a href="#cb23-8" tabindex="-1"></a>         <span class="at">font.legend =</span> <span class="st">&quot;bold&quot;</span>, <span class="at">font.main =</span> <span class="st">&quot;bold&quot;</span></span>
<span id="cb23-9"><a href="#cb23-9" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="figure/analysis.Rmd/maplot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-maplot-1">
Past versions of maplot-1.png
</button>
</p>
<div id="fig-maplot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/ad9ce58ce4495d2e8eebbe4b2e323cf124094cf2/docs/figure/analysis.Rmd/maplot-1.png" target="_blank">ad9ce58</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-07-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="enrichment-analysis" class="section level2">
<h2>Enrichment analysis</h2>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a>all_genes_res <span class="ot">&lt;-</span> <span class="fu">gost</span>(<span class="at">query =</span> genes_biomart<span class="sc">$</span>ensembl_gene_id, <span class="at">organism =</span> <span class="st">&quot;hsapiens&quot;</span>, <span class="at">significant =</span> <span class="cn">TRUE</span>)</span>
<span id="cb24-2"><a href="#cb24-2" tabindex="-1"></a></span>
<span id="cb24-3"><a href="#cb24-3" tabindex="-1"></a><span class="fu">gostplot</span>(all_genes_res, <span class="at">capped =</span> <span class="cn">FALSE</span>, <span class="at">interactive =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="plotly html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-4e30607ffe6a0017e837" style="width:672px;height:480px;"></div>
<script type="application/json" data-for="htmlwidget-4e30607ffe6a0017e837">{"x":{"data":[{"x":[151.82512674060791,151.60341295669051,152.52901419052046],"y":[2.0238916302274292,1.8174262328821491,1.3024093684435445],"text":["CORUM:324 (48) <br> 39S ribosomal subunit, mitochondrial <br> 9.465e-03","CORUM:178 (44) <br> Respiratory chain complex I (holoenzyme), mitochondrial <br> 1.523e-02","CORUM:929 (34) <br> CEN complex <br> 4.984e-02"],"key":["CORUM:324","CORUM:178","CORUM:929"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(102,170,0,1)","opacity":0.80000000000000004,"size":[8.9416090489350069,8.2430909388255813,3.7795275590551185],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(102,170,0,1)"}},"hoveron":"points","set":"SharedDatae56eaee8","name":"CORUM","legendgroup":"CORUM","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[39.392235025830878,39.467551543661941,39.463247742643027,65.146180323035026,39.493374349775443,77.721886900312853,54.272627048739189,54.261867546191894,43.986542613525636,39.482614847228149,39.508437653341659,39.478311046209235,46.254645750495321,43.988694514035096,50.474522649544227,75.406441952135083,39.506285752832198,63.910989430605618],"y":[32.421560664171061,31.845932491604511,31.845932491604511,31.845932491604511,31.748488502575459,22.436201940000345,22.222891533494384,19.964008288872581,17.97067503271575,15.191194095722246,15.191194095722246,14.775257370166347,13.028605080692449,11.09414205961076,7.4719735163632084,2.4592766727523734,2.4501382848685145,2.0836763544097914],"text":["GO:0000244 (1206) <br> spliceosomal tri-snRNP complex assembly <br> 3.788e-33","GO:0000365 (1137) <br> mRNA trans splicing, via spliceosome <br> 1.426e-32","GO:0000353 (1137) <br> formation of quadruple SL/U4/U5/U6 snRNP <br> 1.426e-32","GO:0045291 (1137) <br> mRNA trans splicing, SL addition <br> 1.426e-32","GO:0000387 (1232) <br> spliceosomal snRNP assembly <br> 1.784e-32","GO:0071826 (1626) <br> ribonucleoprotein complex subunit organization <br> 3.663e-23","GO:0022618 (1618) <br> ribonucleoprotein complex assembly <br> 5.986e-23","GO:0022613 (1877) <br> ribonucleoprotein complex biogenesis <br> 1.086e-20","GO:0006396 (3379) <br> RNA processing <br> 1.070e-18","GO:0000377 (1721) <br> RNA splicing, via transesterification reactions with bulged adenosine as nucleophile <br> 6.439e-16","GO:0000398 (1721) <br> mRNA splicing, via spliceosome <br> 6.439e-16","GO:0000375 (1725) <br> RNA splicing, via transesterification reactions <br> 1.678e-15","GO:0008380 (1861) <br> RNA splicing <br> 9.363e-14","GO:0006397 (1910) <br> mRNA processing <br> 8.051e-12","GO:0016071 (2152) <br> mRNA metabolic process <br> 3.373e-08","GO:0065003 (3270) <br> protein-containing complex assembly <br> 3.473e-03","GO:0000395 (204) <br> mRNA 5'-splice site recognition <br> 3.547e-03","GO:0043933 (3456) <br> protein-containing complex organization <br> 8.248e-03"],"key":["GO:0000244","GO:0000365","GO:0000353","GO:0045291","GO:0000387","GO:0071826","GO:0022618","GO:0022613","GO:0006396","GO:0000377","GO:0000398","GO:0000375","GO:0008380","GO:0006397","GO:0016071","GO:0065003","GO:0000395","GO:0043933"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,153,0,1)","opacity":0.80000000000000004,"size":[20.385755183088438,20.248108255352225,20.248108255352225,20.248108255352225,20.43530808642932,21.067015507799969,21.05598874268459,21.384924684227101,22.631041579756186,21.193459666815421,21.193459666815421,21.198609762691959,21.366127149813259,21.423131801307083,21.682462658412994,22.563717490084006,15.546252076782668,22.677165354330711],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(255,153,0,1)"}},"hoveron":"points","set":"SharedDatae56eaee8","name":"GO:BP","legendgroup":"GO:BP","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[30.428239291385459,34.958947488903867,35.858632062092795,28.98400879231902,33.118683589199236,34.961099844342115,37.055341685760361,29.222920245964886,30.146280728974574],"y":[34.332560824513003,34.051611996614099,33.579098045647264,29.776314007647006,29.292981828873611,29.292981828873611,19.203745189904158,3.3684188137293831,2.1647584465109273],"text":["GO:0030532 (1488) <br> small nuclear ribonucleoprotein complex <br> 4.650e-35","GO:0097525 (1479) <br> spliceosomal snRNP complex <br> 8.879e-35","GO:0120114 (1500) <br> Sm-like protein family complex <br> 2.636e-34","GO:0005688 (1048) <br> U6 snRNP <br> 1.674e-30","GO:0046540 (1172) <br> U4/U6 x U5 tri-snRNP complex <br> 5.094e-30","GO:0097526 (1172) <br> spliceosomal tri-snRNP complex <br> 5.094e-30","GO:1990904 (2794) <br> ribonucleoprotein complex <br> 6.255e-20","GO:0005840 (519) <br> ribosome <br> 4.281e-04","GO:0019814 (169) <br> immunoglobulin complex <br> 6.843e-03"],"key":["GO:0030532","GO:0097525","GO:0120114","GO:0005688","GO:0046540","GO:0097526","GO:1990904","GO:0005840","GO:0019814"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(16,150,24,1)","opacity":0.80000000000000004,"size":[20.867647209958118,20.853924562766032,20.885798586979359,20.05575571800728,20.319085282355143,20.319085282355143,22.237301805432381,18.292029116246532,14.911071001458158],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(16,150,24,1)"}},"hoveron":"points","set":"SharedDatae56eaee8","name":"GO:CC","legendgroup":"GO:CC","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[10.623120665711108,8.7745405180081288,2.7575089778713027,2.7424448788795437,10.636032750561187],"y":[31.468491956131221,30.238939549166346,5.7236973942812721,4.1160210877118484,3.3419774128385575],"text":["GO:0030621 (1046) <br> U4 snRNA binding <br> 3.400e-32","GO:0017069 (1219) <br> snRNA binding <br> 5.768e-31","GO:0003735 (451) <br> structural constituent of ribosome <br> 1.889e-06","GO:0003723 (3113) <br> RNA binding <br> 7.656e-05","GO:0030627 (199) <br> pre-mRNA 5'-splice site binding <br> 4.550e-04"],"key":["GO:0030621","GO:0017069","GO:0003735","GO:0003723","GO:0030627"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(220,57,18,1)","opacity":0.80000000000000004,"size":[20.051220568027119,20.410682206224259,17.913198568491161,22.462238508478212,15.46448571130697],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(220,57,18,1)"}},"hoveron":"points","set":"SharedDatae56eaee8","name":"GO:MF","legendgroup":"GO:MF","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[164.95571258424411],"y":[2.555849174998746],"text":"HP:0003287 (176) <br> Abnormality of mitochondrial metabolism <br> 2.781e-03","key":["HP:0003287"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(153,0,153,1)","opacity":0.80000000000000004,"size":15.051059504953191,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(153,0,153,1)"}},"hoveron":"points","set":"SharedDatae56eaee8","name":"HP","legendgroup":"HP","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isSimpleKey":true,"_isNestedKey":false,"frame":null},{"x":[101.5763018678488],"y":[4.0748510861140925],"text":"KEGG:01100 (1532) <br> Metabolic pathways <br> 8.417e-05","key":["KEGG:01100"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(221,68,119,1)","opacity":0.80000000000000004,"size":20.93340978012673,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(221,68,119,1)"}},"hoveron":"points","set":"SharedDatae56eaee8","name":"KEGG","legendgroup":"KEGG","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isSimpleKey":true,"_isNestedKey":false,"frame":null},{"x":[107.43613166518952],"y":[1.781448856167055],"text":"REAC:R-HSA-1430728 (2072) <br> Metabolism <br> 1.654e-02","key":["REAC:R-HSA-1430728"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(51,102,204,1)","opacity":0.80000000000000004,"size":21.600518066729293,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(51,102,204,1)"}},"hoveron":"points","set":"SharedDatae56eaee8","name":"REAC","legendgroup":"REAC","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isSimpleKey":true,"_isNestedKey":false,"frame":null},{"x":[199.07784972238571,199.82547869426975],"y":[1.7399371029010355,1.3141317409919822],"text":["WP:WP111 (103) <br> Electron transport chain: OXPHOS system in mitochondria <br> 1.820e-02","WP:WP4803 (183) <br> Ciliopathies <br> 4.851e-02"],"key":["WP:WP111","WP:WP4803"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,153,198,1)","opacity":0.80000000000000004,"size":[13.034130954508731,15.183968437715892],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(0,153,198,1)"}},"hoveron":"points","set":"SharedDatae56eaee8","name":"WP","legendgroup":"WP","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[2,26.235983263598325],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(220,57,18,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[28.387806335923489,37.087627017334128],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(16,150,24,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[39.239450089659293,99.036461446503282],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(255,153,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[101.18828451882845,102.39976090854752],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(221,68,119,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[104.55158398087268,110.16999402271368],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(51,102,204,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[112.32181709503885,137.38410041841004],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(85,116,166,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[139.5359234907352,145.25977286312013],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(34,170,153,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[147.41159593544529,149.2600119545726],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(102,51,204,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[151.41183502689779,157.6887029288703],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(102,170,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[159.84052600119546,196.15038852361027],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(153,0,153,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[198.30221159593543,200.00215182307232],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(0,153,198,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":29.283520132835207,"r":6.6417600664176026,"b":55.878166138820198,"l":55.193026151930283},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0,210],"tickmode":"array","ticktext":["GO:MF","GO:CC","GO:BP","KEGG","REAC","TF","MIRNA","HPA","CORUM","HP","WP"],"tickvals":[14.117991631799164,32.737716676628807,69.137955768081284,101.79402271368798,107.36078900179319,124.85295875672443,142.39784817692765,148.33580394500893,154.55026897788406,177.99545726240285,199.15218170950388],"categoryorder":"array","categoryarray":["GO:MF","GO:CC","GO:BP","KEGG","REAC","TF","MIRNA","HPA","CORUM","HP","WP"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.6529680365296811,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":10.62681610626816},"tickangle":-45,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.13283520132835203,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-1,40],"tickmode":"array","ticktext":["0","10","20","30","40"],"tickvals":[0,10,20,30,40],"categoryorder":"array","categoryarray":["0","10","20","30","40"],"nticks":null,"ticks":"outside","tickcolor":"rgba(190,190,190,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.66417600664176002,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"annotations":[{"text":"-log10(p-adj)","x":0,"y":0.5,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":13.283520132835205},"xref":"paper","yref":"paper","textangle":-90,"xanchor":"right","yanchor":"center","annotationType":"axis","xshift":-31.54836031548361},{"text":"query_1","x":0.5,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(169,169,169,1)","family":"","size":13.283520132835205},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"}],"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":24.973017849730184,"yanchor":1,"ysizemode":"pixel"}],"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.8897637795275593,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498}},"hovermode":"closest","barmode":"relative","dragmode":"zoom"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"8f0666139ab":{"x":{},"y":{},"text":{},"colour":{},"size":{},"alpha":{},"type":"scatter"},"8f0659110cd2":{"x":{},"xend":{},"y":{},"yend":{}},"8f066e60dcac":{"x":{},"xend":{},"y":{},"yend":{}},"8f061937d8d1":{"x":{},"xend":{},"y":{},"yend":{}},"8f06217b8e46":{"x":{},"xend":{},"y":{},"yend":{}},"8f0636b9a6d6":{"x":{},"xend":{},"y":{},"yend":{}},"8f06567847ab":{"x":{},"xend":{},"y":{},"yend":{}},"8f0672a957e6":{"x":{},"xend":{},"y":{},"yend":{}},"8f064bc9f7d9":{"x":{},"xend":{},"y":{},"yend":{}},"8f0638aeee6e":{"x":{},"xend":{},"y":{},"yend":{}},"8f0664a794d4":{"x":{},"xend":{},"y":{},"yend":{}},"8f06361c1dec":{"x":{},"xend":{},"y":{},"yend":{}}},"cur_data":"8f0666139ab","visdat":{"8f0666139ab":["function (y) ","x"],"8f0659110cd2":["function (y) ","x"],"8f066e60dcac":["function (y) ","x"],"8f061937d8d1":["function (y) ","x"],"8f06217b8e46":["function (y) ","x"],"8f0636b9a6d6":["function (y) ","x"],"8f06567847ab":["function (y) ","x"],"8f0672a957e6":["function (y) ","x"],"8f064bc9f7d9":["function (y) ","x"],"8f0638aeee6e":["function (y) ","x"],"8f0664a794d4":["function (y) ","x"],"8f06361c1dec":["function (y) ","x"]},"highlight":{"on":"plotly_click","off":"plotly_doubleclick","persistent":false,"dynamic":false,"color":null,"selectize":false,"defaultValues":null,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0,"ctGroups":["SharedDatae56eaee8"]},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a><span class="fu">publish_gosttable</span>(all_genes_res, <span class="at">use_colors =</span> <span class="cn">TRUE</span>, <span class="at">show_columns =</span> <span class="fu">c</span>(<span class="st">&quot;source&quot;</span>, <span class="st">&quot;term_name&quot;</span>, <span class="st">&quot;term_size&quot;</span>, <span class="st">&quot;intersection_size&quot;</span>), <span class="at">filename =</span> <span class="cn">NULL</span>)</span></code></pre></div>
<p><img src="figure/analysis.Rmd/gost-table-1.png" width="400px" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-gost-table-1">
Past versions of gost-table-1.png
</button>
</p>
<div id="fig-gost-table-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/53d12da39293a4bfb7d8aa6da6614d9a628b9176/docs/figure/analysis.Rmd/gost-table-1.png" target="_blank">53d12da</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-07-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="genes-of-interest" class="section level2">
<h2>Genes of Interest</h2>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a>genes_of_interest <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;ACAD9/CFAP92&quot;</span>, <span class="st">&quot;ACADM&quot;</span>, <span class="st">&quot;ADORA2A&quot;</span>, <span class="st">&quot;ADRA1D&quot;</span>, <span class="st">&quot;ANKZF1&quot;</span>, <span class="st">&quot;AVPR1B&quot;</span>, <span class="st">&quot;CARMIL2&quot;</span>, <span class="st">&quot;CCDC178&quot;</span>, <span class="st">&quot;CENPF&quot;</span>, <span class="st">&quot;COQ2&quot;</span>, <span class="st">&quot;CR2&quot;</span>, <span class="st">&quot;DCTPP1&quot;</span>, <span class="st">&quot;DNASE1L3&quot;</span>, <span class="st">&quot;DPEP1&quot;</span>, <span class="st">&quot;EN03&quot;</span>, <span class="st">&quot;FCRL3&quot;</span>, <span class="st">&quot;FASTKD1&quot;</span>, <span class="st">&quot;GCKR&quot;</span>, <span class="st">&quot;GIMAP2&quot;</span>, <span class="st">&quot;HAGHL&quot;</span>, <span class="st">&quot;HSD11B1&quot;</span>, <span class="st">&quot;IRF2BP2&quot;</span>, <span class="st">&quot;KCNJ18&quot;</span>, <span class="st">&quot;LRCOL1&quot;</span>, <span class="st">&quot;LRBA, MAB21L2&quot;</span>, <span class="st">&quot;MFN1&quot;</span>, <span class="st">&quot;MRPS18B&quot;</span>, <span class="st">&quot;NLRP12&quot;</span>, <span class="st">&quot;P2RX7&quot;</span>, <span class="st">&quot;PGP&quot;</span>, <span class="st">&quot;PIEZO1&quot;</span>, <span class="st">&quot;PLCG2&quot;</span>, <span class="st">&quot;RERGL&quot;</span>, <span class="st">&quot;RPS6KC1&quot;</span>, <span class="st">&quot;SCN4A&quot;</span>, <span class="st">&quot;SIAE&quot;</span>, <span class="st">&quot;SLC11A2&quot;</span>, <span class="st">&quot;SLC12A3&quot;</span>, <span class="st">&quot;SLC4A1&quot;</span>, <span class="st">&quot;SLC6A12&quot;</span>, <span class="st">&quot;SLC9A9&quot;</span>, <span class="st">&quot;TDO2&quot;</span>, <span class="st">&quot;THEMIS&quot;</span>, <span class="st">&quot;TF&quot;</span>, <span class="st">&quot;TRAFD1&quot;</span>, <span class="st">&quot;UBASH3B&quot;</span>, <span class="st">&quot;WASHC5&quot;</span>)</span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a></span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a>genes_interest_mart <span class="ot">&lt;-</span> <span class="fu">retrieve_gene_info</span>(<span class="at">values =</span> genes_of_interest, <span class="at">filters =</span> <span class="st">&quot;hgnc_symbol&quot;</span>)</span>
<span id="cb26-4"><a href="#cb26-4" tabindex="-1"></a></span>
<span id="cb26-5"><a href="#cb26-5" tabindex="-1"></a>filtered_by_interest <span class="ot">&lt;-</span> <span class="fu">filter</span>(res_aff_vs_unaff_df_genename, Ensembl_ID <span class="sc">%in%</span> genes_interest_mart<span class="sc">$</span>ensembl_gene_id_version)</span>
<span id="cb26-6"><a href="#cb26-6" tabindex="-1"></a></span>
<span id="cb26-7"><a href="#cb26-7" tabindex="-1"></a><span class="fu">paged_table</span>(filtered_by_interest, <span class="at">options =</span> <span class="fu">list</span>(<span class="at">rows.print =</span> <span class="dv">15</span>))</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["gene_name"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Ensembl_ID"],"name":[2],"type":["chr"],"align":["left"]},{"label":["baseMean"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["log2FoldChange"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["lfcSE"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["stat"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["pvalue"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["padj"],"name":[8],"type":["dbl"],"align":["right"]}],"data":[{"1":"SIAE","2":"ENSG00000110013.13","3":"2.466639e+02","4":"0.59408200","5":"0.22599932","6":"2.62868933","7":"0.008571463","8":"0.4327995"},{"1":"SLC9A9","2":"ENSG00000181804.15","3":"6.706541e+02","4":"-0.55536081","5":"0.24074420","6":"-2.30685020","7":"0.021063172","8":"0.5659292"},{"1":"MRPS18B","2":"ENSG00000204568.12","3":"5.667532e+02","4":"-0.31161333","5":"0.14891684","6":"-2.09253244","7":"0.036390913","8":"0.6195764"},{"1":"SLC4A1","2":"ENSG00000004939.16","3":"5.296179e+04","4":"0.90172281","5":"0.56492353","6":"1.59618561","7":"0.110447359","8":"0.7261136"},{"1":"DCTPP1","2":"ENSG00000179958.10","3":"2.499668e+02","4":"-0.21356292","5":"0.13908593","6":"-1.53547470","7":"0.124667268","8":"0.7371881"},{"1":"SCN4A","2":"ENSG00000007314.12","3":"1.847233e+01","4":"0.81871097","5":"0.56661040","6":"1.44492753","7":"0.148478244","8":"0.7615943"},{"1":"RPS6KC1","2":"ENSG00000136643.12","3":"8.229869e+02","4":"-0.17255913","5":"0.12415004","6":"-1.38992411","7":"0.164551923","8":"0.7714993"},{"1":"PIEZO1","2":"ENSG00000103335.22","3":"7.042958e+03","4":"-0.18624466","5":"0.13610317","6":"-1.36840799","7":"0.171184403","8":"0.7739906"},{"1":"TF","2":"ENSG00000091513.16","3":"4.136235e+01","4":"1.00864485","5":"0.79068018","6":"1.27566730","7":"0.202073154","8":"0.7940612"},{"1":"IRF2BP2","2":"ENSG00000168264.11","3":"5.613682e+03","4":"-0.12180823","5":"0.10830348","6":"-1.12469359","7":"0.260718900","8":"0.8269407"},{"1":"DNASE1L3","2":"ENSG00000163687.14","3":"2.668997e+01","4":"0.63701556","5":"0.60275359","6":"1.05684242","7":"0.290583512","8":"0.8382219"},{"1":"COQ2","2":"ENSG00000173085.15","3":"3.108302e+02","4":"0.16249033","5":"0.15633871","6":"1.03934799","7":"0.298642922","8":"0.8427938"},{"1":"HAGHL","2":"ENSG00000103253.19","3":"1.498627e+02","4":"-0.22565536","5":"0.23901399","6":"-0.94410945","7":"0.345113720","8":"0.8653454"},{"1":"ADORA2A","2":"ENSG00000128271.22","3":"1.283026e+03","4":"-0.10698243","5":"0.11661451","6":"-0.91740234","7":"0.358931849","8":"0.8706468"},{"1":"ANKZF1","2":"ENSG00000163516.14","3":"2.089193e+03","4":"-0.09892576","5":"0.10833298","6":"-0.91316382","7":"0.361156387","8":"0.8714518"},{"1":"GIMAP2","2":"ENSG00000106560.11","3":"1.683437e+03","4":"-0.11941041","5":"0.16565308","6":"-0.72084631","7":"0.471004079","8":"0.9046486"},{"1":"ACADM","2":"ENSG00000117054.15","3":"1.035077e+03","4":"0.09260737","5":"0.14470412","6":"0.63997744","7":"0.522187269","8":"0.9169181"},{"1":"HSD11B1","2":"ENSG00000117594.10","3":"4.069608e+00","4":"0.56116575","5":"0.98367616","6":"0.57047815","7":"0.568353441","8":"0.9280564"},{"1":"PGP","2":"ENSG00000184207.9","3":"6.671283e+02","4":"-0.04890482","5":"0.09161347","6":"-0.53381695","7":"0.593468184","8":"0.9338074"},{"1":"P2RX7","2":"ENSG00000089041.17","3":"1.019081e+03","4":"0.09843146","5":"0.19686960","6":"0.49998305","7":"0.617087012","8":"0.9384532"},{"1":"SLC12A3","2":"ENSG00000070915.10","3":"3.199615e+01","4":"-0.28589500","5":"0.59193847","6":"-0.48298094","7":"0.629109270","8":"0.9413566"},{"1":"FASTKD1","2":"ENSG00000138399.18","3":"3.874435e+02","4":"0.08135263","5":"0.17507444","6":"0.46467450","7":"0.642164575","8":"0.9427307"},{"1":"TRAFD1","2":"ENSG00000135148.12","3":"2.902964e+03","4":"0.07679341","5":"0.18963348","6":"0.40495703","7":"0.685509096","8":"0.9527258"},{"1":"FCRL3","2":"ENSG00000160856.21","3":"1.711002e+03","4":"0.13880061","5":"0.35540614","6":"0.39054085","7":"0.696136657","8":"0.9552468"},{"1":"THEMIS","2":"ENSG00000172673.12","3":"1.674831e+03","4":"-0.06364415","5":"0.16796859","6":"-0.37890509","7":"0.704758348","8":"0.9564953"},{"1":"SLC11A2","2":"ENSG00000110911.17","3":"9.435833e+02","4":"0.04955194","5":"0.13329182","6":"0.37175526","7":"0.710075075","8":"0.9578674"},{"1":"CARMIL2","2":"ENSG00000159753.15","3":"1.756139e+03","4":"-0.05190117","5":"0.19826345","6":"-0.26177880","7":"0.793491988","8":"0.9716801"},{"1":"SLC6A12","2":"ENSG00000111181.13","3":"1.512612e+02","4":"-0.07875297","5":"0.34108001","6":"-0.23089294","7":"0.817397976","8":"0.9750350"},{"1":"UBASH3B","2":"ENSG00000154127.10","3":"1.728941e+03","4":"-0.03073140","5":"0.17865576","6":"-0.17201459","7":"0.863426060","8":"0.9809246"},{"1":"WASHC5","2":"ENSG00000164961.16","3":"1.394729e+03","4":"-0.01778942","5":"0.18565470","6":"-0.09581991","7":"0.923663606","8":"0.9908813"},{"1":"MFN1","2":"ENSG00000171109.19","3":"1.262655e+03","4":"0.00887765","5":"0.12420906","6":"0.07147345","7":"0.943020953","8":"0.9923648"},{"1":"GCKR","2":"ENSG00000084734.9","3":"4.139054e-01","4":"0.83560399","5":"2.78469874","6":"0.30006980","7":"0.764123917","8":"NA"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[15],"max":[15]},"pages":{}}}
  </script>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>R version 4.1.1 (2021-08-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] rmarkdown_2.22              ggpubr_0.6.0               
 [3] plotly_4.10.2               biomaRt_2.50.3             
 [5] gprofiler2_0.2.2            limma_3.50.3               
 [7] genefilter_1.76.0           pheatmap_1.0.12            
 [9] RColorBrewer_1.1-3          DESeq2_1.34.0              
[11] SummarizedExperiment_1.24.0 Biobase_2.54.0             
[13] MatrixGenerics_1.6.0        matrixStats_1.0.0          
[15] GenomicRanges_1.46.1        GenomeInfoDb_1.30.1        
[17] IRanges_2.28.0              S4Vectors_0.32.4           
[19] BiocGenerics_0.40.0         lubridate_1.9.2            
[21] forcats_1.0.0               stringr_1.5.0              
[23] dplyr_1.1.2                 purrr_1.0.1                
[25] readr_2.1.4                 tidyr_1.3.0                
[27] tibble_3.2.1                ggplot2_3.4.2.9000         
[29] tidyverse_2.0.0             workflowr_1.7.0            

loaded via a namespace (and not attached):
  [1] colorspace_2.1-0       ggsignif_0.6.4         ellipsis_0.3.2        
  [4] rprojroot_2.0.3        XVector_0.34.0         fs_1.6.2              
  [7] rstudioapi_0.14        farver_2.1.1           ggrepel_0.9.3         
 [10] bit64_4.0.5            AnnotationDbi_1.56.2   fansi_1.0.4           
 [13] xml2_1.3.4             splines_4.1.1          cachem_1.0.8          
 [16] geneplotter_1.72.0     knitr_1.43             jsonlite_1.8.5        
 [19] broom_1.0.5            annotate_1.72.0        dbplyr_2.3.2          
 [22] png_0.1-8              shiny_1.7.4            compiler_4.1.1        
 [25] httr_1.4.6             backports_1.4.1        Matrix_1.5-4.1        
 [28] fastmap_1.1.1          lazyeval_0.2.2         cli_3.6.1             
 [31] later_1.3.1            htmltools_0.5.5        prettyunits_1.1.1     
 [34] tools_4.1.1            gtable_0.3.3           glue_1.6.2            
 [37] GenomeInfoDbData_1.2.7 rappdirs_0.3.3         Rcpp_1.0.10           
 [40] carData_3.0-5          jquerylib_0.1.4        vctrs_0.6.3           
 [43] Biostrings_2.62.0      crosstalk_1.2.0        xfun_0.39             
 [46] ps_1.7.5               mime_0.12              timechange_0.2.0      
 [49] lifecycle_1.0.3        rstatix_0.7.2          XML_3.99-0.14         
 [52] getPass_0.2-2          zlibbioc_1.40.0        scales_1.2.1          
 [55] vroom_1.6.3            hms_1.1.3              promises_1.2.0.1      
 [58] parallel_4.1.1         yaml_2.3.7             curl_5.0.1            
 [61] gridExtra_2.3          memoise_2.0.1          sass_0.4.6            
 [64] stringi_1.7.12         RSQLite_2.3.1          highr_0.10            
 [67] filelock_1.0.2         BiocParallel_1.28.3    rlang_1.1.1           
 [70] pkgconfig_2.0.3        bitops_1.0-7           evaluate_0.21         
 [73] lattice_0.21-8         labeling_0.4.2         htmlwidgets_1.6.2     
 [76] bit_4.0.5              processx_3.8.1         tidyselect_1.2.0      
 [79] magrittr_2.0.3         R6_2.5.1               generics_0.1.3        
 [82] DelayedArray_0.20.0    DBI_1.1.3              pillar_1.9.0          
 [85] whisker_0.4.1          withr_2.5.0            abind_1.4-5           
 [88] survival_3.5-5         KEGGREST_1.34.0        RCurl_1.98-1.12       
 [91] car_3.1-2              crayon_1.5.2           utf8_1.2.3            
 [94] BiocFileCache_2.2.1    tzdb_0.4.0             progress_1.2.2        
 [97] locfit_1.5-9.8         grid_4.1.1             data.table_1.14.8     
[100] blob_1.2.4             callr_3.7.3            git2r_0.32.0          
[103] digest_0.6.32          xtable_1.8-4           httpuv_1.6.11         
[106] munsell_0.5.0          viridisLite_0.4.2      bslib_0.5.0           </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
