<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Differential Gene Expression Analysis</title>

<script src="site_libs/header-attrs-2.22/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">mecfs-dge-analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="background.html">Background</a>
</li>
<li>
  <a href="analysis.html">Analysis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Differential Gene Expression Analysis</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-06-26
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>mecfs-dge-analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230618code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230618)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230618code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230618)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomuabcgdswortheymecfsdgeanalysistree19c181e90936e7e159bfacaaeaf7a602a7940b96targetblank19c181ea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/tree/19c181e90936e7e159bfacaaeaf7a602a7940b96" target="_blank">19c181e</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomuabcgdswortheymecfsdgeanalysistree19c181e90936e7e159bfacaaeaf7a602a7940b96targetblank19c181ea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/tree/19c181e90936e7e159bfacaaeaf7a602a7940b96" target="_blank">19c181e</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    output/batch-correction-limma/

Untracked files:
    Untracked:  Rplot.png
    Untracked:  data/7548-EW_RIN_7548-EW-1-12_WellTable.csv
    Untracked:  data/8289-EW_RIN_8289-EW-1__4__7_WellTable.csv
    Untracked:  data/8921-EW_MA_CHIP_1_Eukaryote_Total_RNA_Pico_8921-EW-1-7.pdf
    Untracked:  data/CPAM0066_PCS_slides_NM_001173543.1(BANP) c.1372G-A, p.Val458Met.pptx
    Untracked:  data/MECFS_RNAseq_metadata_CORRECTED.csv
    Untracked:  data/MECFS_RNAseq_metadata_CORRECTED.xlsx
    Untracked:  output/counts_vst.csv
    Untracked:  output/counts_vst_limma.csv
    Untracked:  output/res_aff_vs_unaff.csv
    Untracked:  output/res_aff_vs_unaff_df_genename_05.csv
    Untracked:  output/res_aff_vs_unaff_genename.csv

Unstaged changes:
    Modified:   analysis/_site.yml
    Modified:   analysis/analysis.Rmd
    Modified:   code/copy_data.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/analysis.Rmd</code>) and HTML
(<code>docs/analysis.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/uab-cgds-worthey/mecfs-dge-analysis/19c181e90936e7e159bfacaaeaf7a602a7940b96/docs/analysis.html" target="_blank">19c181e</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/13c1acc043d29c30392f20ae81dcc31643366350/analysis/analysis.Rmd" target="_blank">13c1acc</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-23
</td>
<td>
wflow_publish("analysis/")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/uab-cgds-worthey/mecfs-dge-analysis/6133d80bd1e4bcfaa02fe0a71a89feec0506e7a2/docs/analysis.html" target="_blank">6133d80</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/df39c694f4c82d2df0924b35abe610451265461c/analysis/analysis.Rmd" target="_blank">df39c69</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-23
</td>
<td>
wflow_publish("analysis/")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse) # Available via CRAN
library(DESeq2) # Available via Bioconductor
library(RColorBrewer) # Available via CRAN
library(pheatmap) # Available via CRAN
library(genefilter) # Available via Bioconductor
library(limma) # Available via Bioconductor</code></pre>
<div id="data-import" class="section level2">
<h2>Data Import</h2>
<p>We will be importing counts data from the star-salmon pipeline and
our metadata for the project which is hosted on Box. This code block
also ensures data is properly ordered by sample id.</p>
<pre class="r"><code>counts &lt;- read_tsv(&quot;data/star-salmon/salmon.merged.gene_counts_length_scaled.tsv&quot;)
# Use first column (gene_id) for row names
counts = data.frame(counts, row.names = 1)
counts$Ensembl_ID = row.names(counts)
drop = c(&quot;Ensembl_ID&quot;,&quot;gene_name&quot;)
gene_names = counts[,drop]    
counts = counts[ , !(names(counts) %in% drop)]     # remove both columns

# Import metadata
sample_metadata &lt;- read_csv(&quot;data/MECFS_RNAseq_metadata_CORRECTED.csv&quot;)
row.names(sample_metadata) &lt;- sample_metadata$RNA_Samples_id
# Order sample_metadata
sample_metadata &lt;- arrange(sample_metadata, RNA_Samples_id)

# Check that data is ordered properly
all(rownames(sample_metadata) %in% colnames(counts))                   </code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code>all(rownames(sample_metadata) == colnames(counts))   </code></pre>
<pre><code>[1] TRUE</code></pre>
</div>
<div id="deseq2-analysis" class="section level2">
<h2>DESeq2 Analysis</h2>
<pre class="r"><code>sample_metadata$Family = factor(sample_metadata$Family)  
sample_metadata$Affected = factor(sample_metadata$Affected) 
sample_metadata$Batch = factor(sample_metadata$Batch)
sample_metadata$Gender = factor(sample_metadata$Gender)

# Account for Family later but batch is accounted for
dds &lt;- DESeqDataSetFromMatrix(countData = round(counts), colData = sample_metadata, design = ~ Batch + Affected)

# Pre-filtering: Keep only rows that have atleast 10 reads total
keep = rowSums(counts(dds)) &gt;= 10 
dds = dds[keep,]

# Run DESeq function
dds = DESeq(dds)
head(counts(dds, normalized=TRUE))</code></pre>
<pre><code>                       LW001974    LW001975     LW001976     LW001977
ENSG00000000003.15     6.283344     2.05296     7.223942     4.264822
ENSG00000000419.14   921.557182  1043.92997   850.017210   859.361676
ENSG00000000457.14  1114.246411   746.25082   829.549373   916.936776
ENSG00000000460.17   234.578192   189.89876   193.842452   328.391311
ENSG00000000938.13 12201.207645 11118.82929 15551.943765 17490.035904
ENSG00000000971.17    83.777926   130.36294   167.354663   197.248027
                       LW001978    LW001979     LW001980    LW001981
ENSG00000000003.15     3.565091    14.47998     4.573315    10.46587
ENSG00000000419.14   910.286661   866.16626   868.015180  1118.80153
ENSG00000000457.14  1100.424867   971.47523  1048.203789  1003.67696
ENSG00000000460.17   294.714219   276.43604   185.676587   252.22747
ENSG00000000938.13 10116.540921 10099.13001 18749.676681 21708.30816
ENSG00000000971.17   553.777525   143.48347   222.263107   216.64352
                      LW001982     LW001983     LW001984     LW001985
ENSG00000000003.15    17.45822     6.161279     5.718447     9.434676
ENSG00000000419.14   979.40627  1059.740056   924.754609   919.356806
ENSG00000000457.14  1104.23256   635.844033   901.880820   939.274457
ENSG00000000460.17   374.47887   144.173938   303.894624   263.122644
ENSG00000000938.13 19973.95218 23219.397520 19682.078435 13836.477178
ENSG00000000971.17   107.36807    27.109629    86.593629   116.361010
                     LW001986     LW001987    LW001988    LW001989     LW001990
ENSG00000000003.15    14.3010     8.261848    20.80204    11.32375     7.592455
ENSG00000000419.14   894.8338   840.642993   896.56804   949.93645   832.458476
ENSG00000000457.14  1046.6967   747.008720   897.60814   859.34649  1079.755586
ENSG00000000460.17   305.7689   232.708707   200.73971   223.95853   264.651294
ENSG00000000938.13 14919.3442 14147.037031 13752.23045 14215.07557 13223.345284
ENSG00000000971.17   222.6869   227.200809   160.17573   163.56522   143.172011
                       LW001991    LW001992     LW001993    LW001994
ENSG00000000003.15     5.130748    3.925784     7.688938    12.43332
ENSG00000000419.14  1002.206201  917.324885   997.639691   920.06541
ENSG00000000457.14   903.866855 1092.676575   897.683498   885.63469
ENSG00000000460.17   217.201685  378.183869   149.934289   273.53296
ENSG00000000938.13 14667.099624 8978.268242 13184.606244 15375.23032
ENSG00000000971.17   149.646830  294.433808   152.817641   124.33316
                       LW001995    LW001996    LW001997   LW001998     LW001999
ENSG00000000003.15     4.865262    13.74314    6.723075   11.82344     3.302019
ENSG00000000419.14  1119.010269   992.12394  808.113622 1212.74707   893.746405
ENSG00000000457.14   904.938740   897.23081  935.852048  728.83058   989.504949
ENSG00000000460.17   515.717776   293.84146  404.729118  143.57034   159.597572
ENSG00000000938.13 15497.481354 14288.28620 9146.071308 8941.89828 16707.114147
ENSG00000000971.17   229.478193   126.96045  208.415327  369.06021   184.913049
                     LW002000    LW002001     LW002080
ENSG00000000003.15   10.49324     9.01276     4.661212
ENSG00000000419.14 1437.57367   736.47124   918.258773
ENSG00000000457.14 1143.76299   987.54098  1041.780892
ENSG00000000460.17  339.71860   350.21010   205.093330
ENSG00000000938.13 8890.39630 10409.73770 12261.318282
ENSG00000000971.17  247.90276   133.90386   209.754542</code></pre>
<pre class="r"><code>resultsNames(dds)</code></pre>
<pre><code>[1] &quot;Intercept&quot;          &quot;Batch_B2_vs_B1&quot;     &quot;Batch_B3_vs_B1&quot;    
[4] &quot;Affected_Yes_vs_No&quot;</code></pre>
<pre class="r"><code># Normalize gene counts for differences in seq. depth/global differences
counts_norm = counts(dds, normalized=TRUE)</code></pre>
<div id="data-transformation-and-visualization" class="section level3">
<h3>Data transformation and visualization</h3>
<p>Perform count data transformation by variance stabilizing
transformation (vst) on normalized counts.</p>
<pre class="r"><code>vsd = vst(dds, blind=FALSE) </code></pre>
<div id="batch-correction-transformation-and-plotting"
class="section level4">
<h4>Batch correction, transformation and plotting</h4>
<pre class="r"><code>counts_vst = assay(vsd)
write.csv(counts_vst, file=&quot;output/counts_vst.csv&quot;)
mm = model.matrix(~ Family + Affected, colData(vsd))

counts_vst_limma = limma::removeBatchEffect(counts_vst, batch=vsd$Batch, design=mm)</code></pre>
<pre><code>Coefficients not estimable: batch2 </code></pre>
<pre class="r"><code>write.csv(counts_vst_limma, file=&quot;output/counts_vst_limma.csv&quot;)

vsd_limma = vsd
assay(vsd_limma) = counts_vst_limma</code></pre>
</div>
<div id="sample-distances-heatmap" class="section level4">
<h4>Sample distances heatmap</h4>
<pre class="r"><code>sampleDists = dist(t(assay(vsd_limma)))
sampleDistMatrix = as.matrix(sampleDists) 
rownames(sampleDistMatrix) = paste(vsd_limma$Batch, vsd_limma$Family, sep=&quot; | &quot;) 
colnames(sampleDistMatrix) = paste(vsd_limma$RNA_Samples_id, vsd_limma$Family, sep=&quot; | &quot;) 
colors = colorRampPalette(rev(brewer.pal(9, &quot;Blues&quot;)))(255) 

pheatmap(sampleDistMatrix, clustering_distance_rows=sampleDists, clustering_distance_cols=sampleDists, col=colors)</code></pre>
<p><img src="figure/analysis.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/6133d80bd1e4bcfaa02fe0a71a89feec0506e7a2/docs/figure/analysis.Rmd/unnamed-chunk-2-1.png" target="_blank">6133d80</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="principal-components-analysis" class="section level4">
<h4>Principal Components Analysis</h4>
<pre class="r"><code>pcaData = plotPCA(vsd, intgroup=c(&quot;Batch&quot;, &quot;Family&quot;, &quot;Affected&quot;), returnData=TRUE) 
percentVar = round(100 * attr(pcaData, &quot;percentVar&quot;)) 

ggplot(pcaData, aes(PC1, PC2, shape=factor(Batch), fill=factor(Affected),  color=factor(Family)))  + 
  geom_point(size=5) + xlab(paste0(&quot;PC1: &quot;,percentVar[1],&quot;% variance&quot;)) + ylab(paste0(&quot;PC2: &quot;,percentVar[2],&quot;% variance&quot;)) + coord_fixed()</code></pre>
<p><img src="figure/analysis.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/19c181e90936e7e159bfacaaeaf7a602a7940b96/docs/figure/analysis.Rmd/unnamed-chunk-3-1.png" target="_blank">19c181e</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/6133d80bd1e4bcfaa02fe0a71a89feec0506e7a2/docs/figure/analysis.Rmd/unnamed-chunk-3-1.png" target="_blank">6133d80</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(pcaData, aes(PC1, PC2, shape=factor(Batch), color=factor(Affected)))  + 
  geom_point(size=5) + xlab(paste0(&quot;PC1: &quot;,percentVar[1],&quot;% variance&quot;)) + ylab(paste0(&quot;PC2: &quot;,percentVar[2],&quot;% variance&quot;)) + coord_fixed()</code></pre>
<p><img src="figure/analysis.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-2">
Past versions of unnamed-chunk-3-2.png
</button>
</p>
<div id="fig-unnamed-chunk-3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/19c181e90936e7e159bfacaaeaf7a602a7940b96/docs/figure/analysis.Rmd/unnamed-chunk-3-2.png" target="_blank">19c181e</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="heatmap-of-top-50-top-100-genes" class="section level4">
<h4>Heatmap of top 50 &amp; top 100 genes</h4>
<p>This is a heatmap for 50 genes with the highest variance across
samples</p>
<pre class="r"><code>topVarGenes = head(order(-rowVars(assay(vsd))),50)
mat = assay(vsd_limma)[ topVarGenes, ]
mat = mat - rowMeans(mat)
df = as.data.frame(colData(vsd)[,c(&quot;Batch&quot;, &quot;Affected&quot;)])
pheatmap(mat, annotation_col=df, fontsize = 5)</code></pre>
<p><img src="figure/analysis.Rmd/top50-heatmap-1.png" width="800px" height="570px" style="display: block; margin: auto;" /></p>
<p>This is a heatmap of the top 100 genes with the highest variance
across samples</p>
<pre class="r"><code>topVarGenes = head(order(-rowVars(assay(vsd_limma))),100)
mat = assay(vsd_limma)[ topVarGenes, ]
mat = mat - rowMeans(mat)
df = as.data.frame(colData(vsd_limma)[,c(&quot;Batch&quot;, &quot;Family&quot;, &quot;Affected&quot;)])
pheatmap(mat, annotation_col=df, fontsize = 6)</code></pre>
<p><img src="figure/analysis.Rmd/top100-heatmap-1.png" width="1200px" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisoncontrast-of-affected_yes_vs_no"
class="section level4">
<h4>Comparison/Contrast of Affected_Yes_vs_No</h4>
<pre class="r"><code>res_aff_vs_unaff = results(dds, contrast=c(&quot;Affected&quot;, &quot;Yes&quot;, &quot;No&quot;))
res_aff_vs_unaff= res_aff_vs_unaff[order(res_aff_vs_unaff$padj),]
summary(res_aff_vs_unaff)</code></pre>
<pre><code>
out of 29624 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 34, 0.11%
LFC &lt; 0 (down)     : 30, 0.1%
outliers [1]       : 162, 0.55%
low counts [2]     : 6, 0.02%
(mean count &lt; 0)
[1] see &#39;cooksCutoff&#39; argument of ?results
[2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<pre class="r"><code>write.csv(res_aff_vs_unaff, file=&quot;output/res_aff_vs_unaff.csv&quot;)      
res_aff_vs_unaff_df = as.data.frame(res_aff_vs_unaff)
res_aff_vs_unaff_05 = subset(res_aff_vs_unaff_df, padj &lt; 0.05)                             </code></pre>
<pre class="r"><code>topgenes_byensemblid = head(rownames(res_aff_vs_unaff_05),50)
topgenes_aff_vs_unaff_05 = assay(vsd_limma)[topgenes_byensemblid,] 
topgenes_aff_vs_unaff_05 = topgenes_aff_vs_unaff_05 - rowMeans(topgenes_aff_vs_unaff_05)

# Convert ensemblids
ensemblids &lt;- topgenes_byensemblid
rownames(topgenes_aff_vs_unaff_05) &lt;- gene_names$gene_name[match(ensemblids, gene_names$Ensembl_ID)]

df = as.data.frame(colData(vsd_limma)[,c(&quot;Batch&quot;, &quot;Family&quot;, &quot;Affected&quot;)])
pheatmap(topgenes_aff_vs_unaff_05, annotation_col=df, fontsize = 5)</code></pre>
<p><img src="figure/analysis.Rmd/top%2050%20genes-1.png" width="800px" height="570px" style="display: block; margin: auto;" /></p>
<pre class="r"><code>res_aff_vs_unaff_df_genename = res_aff_vs_unaff_df
res_aff_vs_unaff_df_genename$Ensembl_ID = row.names(res_aff_vs_unaff_df)
res_aff_vs_unaff_df_genename = merge(x=res_aff_vs_unaff_df_genename, y=gene_names, by.x =&quot;Ensembl_ID&quot;, by.y=&quot;Ensembl_ID&quot;, all.x=T)
res_aff_vs_unaff_df_genename = res_aff_vs_unaff_df_genename[,c(dim(res_aff_vs_unaff_df_genename)[2],1:dim(res_aff_vs_unaff_df_genename)[2]-1)]
res_aff_vs_unaff_df_genename = res_aff_vs_unaff_df_genename[order(res_aff_vs_unaff_df_genename[,&quot;padj&quot;]),]
write.csv(res_aff_vs_unaff_df_genename,file=&quot;output/res_aff_vs_unaff_genename.csv&quot; )</code></pre>
<pre class="r"><code>res_aff_vs_unaff_df_genename_05= subset(res_aff_vs_unaff_df_genename, padj &lt; 0.05)                   
res_aff_vs_unaff_df_genename_05 = res_aff_vs_unaff_df_genename_05[order(res_aff_vs_unaff_df_genename_05$padj),]
write.csv(res_aff_vs_unaff_df_genename_05, file=&quot;output/res_aff_vs_unaff_df_genename_05.csv&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.1 (2021-08-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] limma_3.50.3                genefilter_1.76.0          
 [3] pheatmap_1.0.12             RColorBrewer_1.1-3         
 [5] DESeq2_1.34.0               SummarizedExperiment_1.24.0
 [7] Biobase_2.54.0              MatrixGenerics_1.6.0       
 [9] matrixStats_1.0.0           GenomicRanges_1.46.1       
[11] GenomeInfoDb_1.30.1         IRanges_2.28.0             
[13] S4Vectors_0.32.4            BiocGenerics_0.40.0        
[15] lubridate_1.9.2             forcats_1.0.0              
[17] stringr_1.5.0               dplyr_1.1.2                
[19] purrr_1.0.1                 readr_2.1.4                
[21] tidyr_1.3.0                 tibble_3.2.1               
[23] ggplot2_3.4.2               tidyverse_2.0.0            
[25] workflowr_1.7.0            

loaded via a namespace (and not attached):
 [1] bitops_1.0-7           fs_1.6.2               bit64_4.0.5           
 [4] httr_1.4.6             rprojroot_2.0.3        tools_4.1.1           
 [7] bslib_0.5.0            utf8_1.2.3             R6_2.5.1              
[10] DBI_1.1.3              colorspace_2.1-0       withr_2.5.0           
[13] tidyselect_1.2.0       processx_3.8.1         bit_4.0.5             
[16] compiler_4.1.1         git2r_0.32.0           cli_3.6.1             
[19] DelayedArray_0.20.0    labeling_0.4.2         sass_0.4.6            
[22] scales_1.2.1           callr_3.7.3            digest_0.6.31         
[25] rmarkdown_2.22         XVector_0.34.0         pkgconfig_2.0.3       
[28] htmltools_0.5.5        highr_0.10             fastmap_1.1.1         
[31] rlang_1.1.1            rstudioapi_0.14        RSQLite_2.3.1         
[34] farver_2.1.1           jquerylib_0.1.4        generics_0.1.3        
[37] jsonlite_1.8.5         vroom_1.6.3            BiocParallel_1.28.3   
[40] RCurl_1.98-1.12        magrittr_2.0.3         GenomeInfoDbData_1.2.7
[43] Matrix_1.5-4.1         Rcpp_1.0.10            munsell_0.5.0         
[46] fansi_1.0.4            lifecycle_1.0.3        stringi_1.7.12        
[49] whisker_0.4.1          yaml_2.3.7             zlibbioc_1.40.0       
[52] grid_4.1.1             blob_1.2.4             parallel_4.1.1        
[55] promises_1.2.0.1       crayon_1.5.2           lattice_0.21-8        
[58] splines_4.1.1          Biostrings_2.62.0      annotate_1.72.0       
[61] hms_1.1.3              KEGGREST_1.34.0        locfit_1.5-9.8        
[64] knitr_1.43             ps_1.7.5               pillar_1.9.0          
[67] geneplotter_1.72.0     XML_3.99-0.14          glue_1.6.2            
[70] evaluate_0.21          getPass_0.2-2          png_0.1-8             
[73] vctrs_0.6.3            tzdb_0.4.0             httpuv_1.6.11         
[76] gtable_0.3.3           cachem_1.0.8           xfun_0.39             
[79] xtable_1.8-4           later_1.3.1            survival_3.5-5        
[82] AnnotationDbi_1.56.2   memoise_2.0.1          timechange_0.2.0      </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
