<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Background</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />
<link rel="preconnect" href="https://fonts.googleapis.com">
<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
<link href="https://fonts.googleapis.com/css2?family=Arimo:ital,wght@0,400..700;1,400..700&family=Noto+Sans:ital,wght@0,100..900;1,100..900&family=Roboto:ital,wght@0,100..900;1,100..900&display=swap" rel="stylesheet">

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="custom.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">mecfs-dge-analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="background.html">Background</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="subsetted-condition-analysis.html">Subsetted Analysis</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="contact.html">Contact</a>
</li>
<li>
  <a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Background</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-07-15
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>mecfs-dge-analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomuabcgdswortheymecfsdgeanalysistree1373f8e34f1d048e21230fe2d9d890856d36c89ctargetblank1373f8ea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/tree/1373f8e34f1d048e21230fe2d9d890856d36c89c" target="_blank">1373f8e</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomuabcgdswortheymecfsdgeanalysistree1373f8e34f1d048e21230fe2d9d890856d36c89ctargetblank1373f8ea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/tree/1373f8e34f1d048e21230fe2d9d890856d36c89c" target="_blank">1373f8e</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .github/.DS_Store
    Ignored:    .test/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/head.html
    Ignored:    configs/
    Ignored:    data/.DS_Store
    Ignored:    notebooks/
    Ignored:    output/.DS_Store
    Ignored:    output/all-samples-analysis/.DS_Store
    Ignored:    output/all-samples-analysis/all-condition-analysis.RData
    Ignored:    output/all-samples-analysis/batch-correction-limma/
    Ignored:    output/publication-analysis/.DS_Store
    Ignored:    output/publication-analysis/batch-correction-limma/plot-counts/padj-05/
    Ignored:    output/publication-analysis/batch-correction-limma/plot-counts/spta1-genes-of-interest/
    Ignored:    output/spta1-analysis/batch-correction-limma/plot-counts/spta1-genes-of-interest/
    Ignored:    renv/.DS_Store
    Ignored:    renv/library/
    Ignored:    renv/staging/
    Ignored:    src/

Untracked files:
    Untracked:  output/all-samples-analysis/Supplemental Figure 3. Genes of Interest Heatmap.png
    Untracked:  output/all-samples-analysis/missing_genes_interest_counts.csv
    Untracked:  output/publication-analysis/Supplemental Figure 4. Significant DEGs (Padj < 0.05).png
    Untracked:  output/publication-analysis/Supplemental Figure 4. Significant DEGs (Padj < 0.1).png.png
    Untracked:  output/publication-analysis/mecfs_deg_variant_network.png
    Untracked:  output/publication-analysis/missing_genes_interest_counts.csv
    Untracked:  output/subgroup-chrn-analysis/
    Untracked:  output/subgroup-hpp-analysis/
    Untracked:  output/subgroup-ion_channel-analysis/
    Untracked:  output/subgroup-mito-analysis/
    Untracked:  output/subgroup-rbc-analysis/

Unstaged changes:
    Deleted:    analysis/spta1-analysis.Rmd
    Modified:   code/helpers.R
    Modified:   data/Prioritized_Genes_From_WGS_2025_06_18.csv
    Modified:   output/all-samples-analysis/all-genes-heatmap.png
    Modified:   output/all-samples-analysis/analysis-volcano-plot.png
    Modified:   output/all-samples-analysis/counts_vst.csv
    Modified:   output/all-samples-analysis/counts_vst_limma.csv
    Modified:   output/all-samples-analysis/genes_of_interest_heatmap.png
    Modified:   output/all-samples-analysis/genes_of_interest_plot_counts_signals.csv
    Modified:   output/all-samples-analysis/madd_genes_of_interest.csv
    Modified:   output/all-samples-analysis/madd_genes_of_interest_heatmap.png
    Modified:   output/all-samples-analysis/mito_genes_of_interest.csv
    Modified:   output/all-samples-analysis/mito_genes_of_interest_heatmap.png
    Modified:   output/all-samples-analysis/mito_genes_of_interest_plot_counts_signals.csv
    Modified:   output/all-samples-analysis/other_genes_of_interest.csv
    Modified:   output/all-samples-analysis/res_aff_vs_unaff.csv
    Modified:   output/all-samples-analysis/res_aff_vs_unaff_df_genename.csv
    Modified:   output/all-samples-analysis/res_aff_vs_unaff_significant_mygene.csv
    Modified:   output/all-samples-analysis/res_aff_vs_unaff_significant_samples.csv
    Modified:   output/all-samples-analysis/res_spta1_vs_unaff_df_genename_05.csv
    Modified:   output/all-samples-analysis/res_spta1_vs_unaff_df_genename_padj05_lfc1.csv
    Modified:   output/all-samples-analysis/res_spta1_vs_unaff_df_genename_padj1.csv
    Modified:   output/all-samples-analysis/res_spta1_vs_unaff_df_genename_padj1_lfc1.csv
    Modified:   output/all-samples-analysis/significant_plot_counts_signals.csv
    Modified:   output/all-samples-analysis/slc4a1_genes_of_interest.csv
    Modified:   output/all-samples-analysis/slc4a1_genes_of_interest_heatmap.png
    Modified:   output/all-samples-analysis/slc4a1_genes_of_interest_plot_counts_signals.csv
    Modified:   output/all-samples-analysis/stringdb_combined_enrichment.csv
    Modified:   output/publication-analysis/Supplemental Figure 4. DEGs (Padj < 0.1).png
    Deleted:    output/publication-analysis/Supplemental Figure 4. Significant DEGs.png
    Modified:   output/publication-analysis/Supplemental Table 5. Significant Differentially Expressed Genes.csv
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/ACACB-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/ACADM-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/ACADVL-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/ADM2-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/ADORA2A-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/ADORA2B-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/ADRB2-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/ATP1A1-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/CALCB-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/CCR5-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/CHRFAM7A-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/COL6A3-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/COMT-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/CPT1A-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/CPT1B-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/CPT1C-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/CUBN-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/CYTH2-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/DENND1A-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/DMXL2-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/ELOVL4-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/ENO3-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/ETFB-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/FKBP5-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/G6PD-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/GABRB3-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/GIPR-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/GLYCTK-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/GPNMB-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/HADHA-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/HSPA1A-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/HTR6-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/IDO1-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/IFNG-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/IL10-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/IL6-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/KLHL7-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/LMBRD1-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/MADD-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/MAPK1-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/MCCC1-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/MCCC2-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/MINPP1-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/MLYCD-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/MMAA-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/MMACHC-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/MYH9-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/NLRP3-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/NR3C1-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/NUP42-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/OXTR-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/P2RX7-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/PCK2-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/PGM1-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/POLG-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/PTPN22-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/SLC12A3-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/SLC25A20-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/SLC7A9-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/SMCR8-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/SURF1-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/SYT6-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/TACO1-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/TNF-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/TOMM7-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/UCP2-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/VIPR2-subsetted-plot-counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/faceted_genes_of_interest_plot_counts.png
    Modified:   output/publication-analysis/batch-correction-limma/plot-counts/genes-of-interest/genes_of_interest_plot_counts_signals.csv
    Modified:   output/publication-analysis/counts_vst_limma_subsetted.csv
    Modified:   output/publication-analysis/counts_vst_subsetted.csv
    Modified:   output/publication-analysis/publication-analysis.RData
    Modified:   output/publication-analysis/res_aff_vs_unaff_df_sub_genename.csv
    Modified:   output/publication-analysis/res_aff_vs_unaff_df_sub_genename_05.csv
    Modified:   output/publication-analysis/res_aff_vs_unaff_df_sub_genename_padj1.csv
    Modified:   output/publication-analysis/res_aff_vs_unaff_df_sub_genename_padj1_lfc1.csv
    Modified:   output/publication-analysis/res_aff_vs_unaff_significant_subsetted_mygene.csv
    Modified:   output/publication-analysis/res_aff_vs_unaff_significant_subsetted_samples.csv
    Modified:   output/publication-analysis/res_aff_vs_unaff_sub.csv
    Modified:   output/publication-analysis/subsetted-analysis-volcano-plot.png
    Modified:   output/spta1-analysis/spta1-condition-analysis.RData

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/background.Rmd</code>) and HTML
(<code>docs/background.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/1373f8e34f1d048e21230fe2d9d890856d36c89c/analysis/background.Rmd" target="_blank">1373f8e</a>
</td>
<td>
sdhutchins
</td>
<td>
2025-07-15
</td>
<td>
Changes to Category.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/8e529df0c9dc12180f1b1c7e74618eb3e58ab946/analysis/background.Rmd" target="_blank">8e529df</a>
</td>
<td>
sdhutchins
</td>
<td>
2024-12-23
</td>
<td>
Remove unused section.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/01889df0642bdb81bf6ee7369da6e24298467dd2/analysis/background.Rmd" target="_blank">01889df</a>
</td>
<td>
Brandon M. Wilk
</td>
<td>
2024-12-06
</td>
<td>
Remove archived analyses.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/uab-cgds-worthey/mecfs-dge-analysis/05f55021fba95b95fe55b566cf65f894df83adb1/docs/background.html" target="_blank">05f5502</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-08-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/uab-cgds-worthey/mecfs-dge-analysis/482a8a9e9f9415f588b75808fea1ca56fb13314f/docs/background.html" target="_blank">482a8a9</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-08-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/uab-cgds-worthey/mecfs-dge-analysis/752cbf3014ebc95eaa3fc70e23f7db3c58c63654/docs/background.html" target="_blank">752cbf3</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-07-27
</td>
<td>
Update metadata.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/uab-cgds-worthey/mecfs-dge-analysis/a3635c99af4510e4caba43ff8c9cca2ae0cac19b/docs/background.html" target="_blank">a3635c9</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-07-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/uab-cgds-worthey/mecfs-dge-analysis/81e7d95f329afff8b850807bc075a0a4b3facc56/docs/background.html" target="_blank">81e7d95</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-07-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/uab-cgds-worthey/mecfs-dge-analysis/e555cb26bda7b6c531332c710b551b03374a88e2/docs/background.html" target="_blank">e555cb2</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-07-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/uab-cgds-worthey/mecfs-dge-analysis/0b76a235f80e48093d721754e30162f4eef00e7b/docs/background.html" target="_blank">0b76a23</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-07-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/uab-cgds-worthey/mecfs-dge-analysis/fad56f5bbc2d6f73cef2a7b2bbbab1754452970d/docs/background.html" target="_blank">fad56f5</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-07-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/uab-cgds-worthey/mecfs-dge-analysis/ae8df0ec58683a6f7764e9a20cbc61eab6878a30/docs/background.html" target="_blank">ae8df0e</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-07-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/uab-cgds-worthey/mecfs-dge-analysis/53d12da39293a4bfb7d8aa6da6614d9a628b9176/docs/background.html" target="_blank">53d12da</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-07-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/uab-cgds-worthey/mecfs-dge-analysis/ad9ce58ce4495d2e8eebbe4b2e323cf124094cf2/docs/background.html" target="_blank">ad9ce58</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-07-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/uab-cgds-worthey/mecfs-dge-analysis/597034daf6d9bfa65ba200fa716e748f8360d8be/docs/background.html" target="_blank">597034d</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/uab-cgds-worthey/mecfs-dge-analysis/08f632068d1a715b373d662057f4d57adbcbe5a0/docs/background.html" target="_blank">08f6320</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/uab-cgds-worthey/mecfs-dge-analysis/37ab1642bb81d2294fecd18f6c34674682fe7f11/docs/background.html" target="_blank">37ab164</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/300901630f5013adb2b8a0b2906eeaa62e2d20c0/analysis/background.Rmd" target="_blank">3009016</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-28
</td>
<td>
Add gprofiler.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/uab-cgds-worthey/mecfs-dge-analysis/b66bc8612aeecb1b8ea4b8bb97c2eb0b6497d622/docs/background.html" target="_blank">b66bc86</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/uab-cgds-worthey/mecfs-dge-analysis/6133d80bd1e4bcfaa02fe0a71a89feec0506e7a2/docs/background.html" target="_blank">6133d80</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/uab-cgds-worthey/mecfs-dge-analysis/blob/df39c694f4c82d2df0924b35abe610451265461c/analysis/background.Rmd" target="_blank">df39c69</a>
</td>
<td>
sdhutchins
</td>
<td>
2023-06-23
</td>
<td>
wflow_publish("analysis/")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="project-background" class="section level1">
<h1>Project Background</h1>
<p>Myalgic Encephalomyelitis/Chronic Fatigue Syndrome (ME/CFS) is a
chronic, disabling illness affecting an estimated 1–3 million Americans.
It is marked by persistent fatigue, post-exertional malaise, cognitive
impairment, pain, and a range of immune, metabolic, and neurological
symptoms.</p>
<p>Onset often follows infection or occurs gradually in adulthood.
Research has identified biological abnormalities in ME/CFS, including
immune dysregulation, altered cytokine profiles, metabolic dysfunction,
and neuroinflammation.</p>
<p>Family studies suggest a genetic component, with first-degree
relatives of ME/CFS patients at increased risk.</p>
</div>
<div id="methods" class="section level1">
<h1>Methods</h1>
<div id="bulk-rna-seq-analysis" class="section level2">
<h2>Bulk RNA-Seq Analysis</h2>
<p>Peripheral blood RNA was collected from 29 ME/CFS patients for
transcriptomic profiling. Bulk RNA sequencing was performed by
Vantage.</p>
<p>We used the <a href="https://nf-co.re/rnaseq">nf-core/rnaseq</a>
pipeline for secondary analysis. This pipeline was modified to run on
UAB’s Cheaha high-performance computing cluster.</p>
<div class="float">
<img
src="https://raw.githubusercontent.com/nf-core/rnaseq/3.12.0/docs/images/nf-core-rnaseq_metro_map_grey.png"
alt="nf-core rnaseq map" />
<div class="figcaption">nf-core rnaseq map</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
