---
title: "Initial Differential Gene Expression (DGE) Analysis"
output:
  workflowr::wflow_html:
    toc: true
    toc_depth: 3
    toc_float: true
    code_folding: hide
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
```


```{r, load libraries}
library(tidyverse) # Available via CRAN
library(DESeq2) # Available via Bioconductor
library(RColorBrewer) # Available via CRAN
library(pheatmap) # Available via CRAN
library(genefilter) # Available via Bioconductor
library(limma) # Available via Bioconductor
```

## Data Import

We will be importing counts data from the star-salmon pipeline and our metadata 
for the project which is hosted on Box. This code block also ensures data is properly ordered
by sample id.

```{r, import data}
counts <- read_tsv("data/star-salmon/salmon.merged.gene_counts_length_scaled.tsv")
# Use first column (gene_id) for row names
counts <- data.frame(counts, row.names = 1)
counts$Ensembl_ID <- row.names(counts)
drop <- c("Ensembl_ID", "gene_name")
gene_names <- counts[, drop]
counts <- counts[, !(names(counts) %in% drop)] # remove both columns

# Import metadata
sample_metadata <- read_csv("data/MECFS_RNAseq_metadata_2023_06_23.csv")
row.names(sample_metadata) <- sample_metadata$RNA_Samples_id
# Order sample_metadata
sample_metadata <- arrange(sample_metadata, RNA_Samples_id)

# Check that data is ordered properly
all(rownames(sample_metadata) %in% colnames(counts))
all(rownames(sample_metadata) == colnames(counts))
```

## DESeq2 Analysis

```{r, start analysis}
sample_metadata$Family <- factor(sample_metadata$Family)
sample_metadata$Affected <- factor(sample_metadata$Affected)
sample_metadata$Batch <- factor(sample_metadata$Batch)
sample_metadata$Gender <- factor(sample_metadata$Gender)

# Account for Family later but batch is accounted for
dds <- DESeqDataSetFromMatrix(countData = round(counts), colData = sample_metadata, design = ~ Batch + Affected)

# Pre-filtering: Keep only rows that have atleast 10 reads total
keep <- rowSums(counts(dds)) >= 10
dds <- dds[keep, ]

# Run DESeq function
dds <- DESeq(dds)
head(counts(dds, normalized = TRUE))
resultsNames(dds)

# Normalize gene counts for differences in seq. depth/global differences
counts_norm <- counts(dds, normalized = TRUE)
```

### Data transformation and visualization

Perform count data transformation by variance stabilizing transformation (vst) on normalized counts.

```{r, vsd}
vsd <- vst(dds, blind = FALSE)
```

#### Batch correction, transformation and plotting

```{r}
counts_vst <- assay(vsd)
write.csv(counts_vst, file = "output/counts_vst.csv")
mm <- model.matrix(~ Family + Affected, colData(vsd))

counts_vst_limma <- limma::removeBatchEffect(counts_vst, batch = vsd$Batch, design = mm)

write.csv(counts_vst_limma, file = "output/counts_vst_limma.csv")

vsd_limma <- vsd
assay(vsd_limma) <- counts_vst_limma
```

#### Sample distances heatmap
```{r}
sampleDists <- dist(t(assay(vsd_limma)))
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste(vsd_limma$Batch, vsd_limma$Family, sep = " | ")
colnames(sampleDistMatrix) <- paste(vsd_limma$RNA_Samples_id, vsd_limma$Family, sep = " | ")
colors <- colorRampPalette(rev(brewer.pal(9, "Blues")))(255)

pheatmap(sampleDistMatrix, clustering_distance_rows = sampleDists, clustering_distance_cols = sampleDists, col = colors)
```


#### Principal Components Analysis

```{r}
pcaData <- plotPCA(vsd, intgroup = c("Batch", "Family", "Affected"), returnData = TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))

ggplot(pcaData, aes(PC1, PC2, shape = factor(Batch), fill = factor(Affected), color = factor(Family))) +
  geom_point(size = 5) +
  xlab(paste0("PC1: ", percentVar[1], "% variance")) +
  ylab(paste0("PC2: ", percentVar[2], "% variance")) +
  coord_fixed()

ggplot(pcaData, aes(PC1, PC2, shape = factor(Batch), color = factor(Affected))) +
  geom_point(size = 5) +
  xlab(paste0("PC1: ", percentVar[1], "% variance")) +
  ylab(paste0("PC2: ", percentVar[2], "% variance")) +
  coord_fixed()
```

#### Heatmap of top 50  & top 100 genes

This is a heatmap for 50 genes with the highest variance across samples 
```{r, top50-heatmap, out.height="570px", out.width ="800px"}
topVarGenes <- head(order(-rowVars(assay(vsd))), 50)
mat <- assay(vsd_limma)[topVarGenes, ]
mat <- mat - rowMeans(mat)
df <- as.data.frame(colData(vsd)[, c("Batch", "Affected")])
pheatmap(mat, annotation_col = df, fontsize = 5)
```

This is a heatmap of the top 100 genes with the highest variance across samples 
```{r, top100-heatmap, out.width = "1200px"}
topVarGenes <- head(order(-rowVars(assay(vsd_limma))), 100)
mat <- assay(vsd_limma)[topVarGenes, ]
mat <- mat - rowMeans(mat)
df <- as.data.frame(colData(vsd_limma)[, c("Batch", "Family", "Affected")])
pheatmap(mat, annotation_col = df, fontsize = 6)
```

#### Comparison/Contrast of Affected_Yes_vs_No 
```{r}
res_aff_vs_unaff <- results(dds, contrast = c("Affected", "Yes", "No"))
res_aff_vs_unaff <- res_aff_vs_unaff[order(res_aff_vs_unaff$padj), ]
summary(res_aff_vs_unaff)
write.csv(res_aff_vs_unaff, file = "output/res_aff_vs_unaff.csv")
res_aff_vs_unaff_df <- as.data.frame(res_aff_vs_unaff)
res_aff_vs_unaff_05 <- subset(res_aff_vs_unaff_df, padj < 0.05)
```

```{r, top 50 genes, out.height="570px", out.width ="800px"}
topgenes_byensemblid <- head(rownames(res_aff_vs_unaff_05), 50)
topgenes_aff_vs_unaff_05 <- assay(vsd_limma)[topgenes_byensemblid, ]
topgenes_aff_vs_unaff_05 <- topgenes_aff_vs_unaff_05 - rowMeans(topgenes_aff_vs_unaff_05)

# Convert ensemblids
ensemblids <- topgenes_byensemblid
ensemblids
rownames(topgenes_aff_vs_unaff_05) <- gene_names$gene_name[match(ensemblids, gene_names$Ensembl_ID)]

df <- as.data.frame(colData(vsd_limma)[, c("Batch", "Family", "Affected")])
pheatmap(topgenes_aff_vs_unaff_05, annotation_col = df, fontsize = 5)
```

```{r, all-genes}
res_aff_vs_unaff_df_genename <- res_aff_vs_unaff_df
res_aff_vs_unaff_df_genename$Ensembl_ID <- row.names(res_aff_vs_unaff_df)
res_aff_vs_unaff_df_genename <- merge(x = res_aff_vs_unaff_df_genename, y = gene_names, by.x = "Ensembl_ID", by.y = "Ensembl_ID", all.x = T)
res_aff_vs_unaff_df_genename <- res_aff_vs_unaff_df_genename[, c(dim(res_aff_vs_unaff_df_genename)[2], 1:dim(res_aff_vs_unaff_df_genename)[2] - 1)]
res_aff_vs_unaff_df_genename <- res_aff_vs_unaff_df_genename[order(res_aff_vs_unaff_df_genename[, "padj"]), ]
write.csv(res_aff_vs_unaff_df_genename, file = "output/res_aff_vs_unaff_genename.csv")
```

```{r, top50-with-genename-padj}
res_aff_vs_unaff_df_genename_05 <- subset(res_aff_vs_unaff_df_genename, padj < 0.05)
res_aff_vs_unaff_df_genename_05 <- res_aff_vs_unaff_df_genename_05[order(res_aff_vs_unaff_df_genename_05$padj), ]
write.csv(res_aff_vs_unaff_df_genename_05, file = "output/res_aff_vs_unaff_df_genename_05.csv")
```
