---
title: "Differential Gene Expression (DGE) Analysis"
output:
  workflowr::wflow_html:
    toc: true
    toc_depth: 3
    toc_float: true
    code_folding: hide
editor_options:
  chunk_output_type: console
---

```{r, setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
```

# DGE Analysis Setup

Ensure you have all necessary libraries installed and load the helper code.

At a later date, `renv` will be integrated to ensure reproducibility of this analysis.

Use the below code to install these packages:

```
# Install packages from CRAN
install.packages(c("tidyverse", "RColorBrewer", "pheatmap", "gprofiler2", "plotly", "ggupset"))

# Install packages from Bioconductor
install.packages("BiocManager")
BiocManager::install(c("DESeq2", "genefilter", "limma", "biomaRt", "mygene"))
```

```{r, load libraries, warning = FALSE, message = FALSE}
library(tidyverse) # Available via CRAN
library(DESeq2) # Available via Bioconductor
library(RColorBrewer) # Available via CRAN
library(pheatmap) # Available via CRAN
library(genefilter) # Available via Bioconductor
library(limma) # Available via Bioconductor
library(gprofiler2) # Available via CRAN
library(biomaRt) # Available via Bioconductor
library(plotly) # Available via CRAN
library(ggpubr)
library(rmarkdown)
library(ggupset)
library(clusterProfiler)
library(DOSE)
library(org.Hs.eg.db) # Available via Bioconductor
library(UpSetR)
library(ggrepel)
library(ReactomePA)
```

```{r, include=FALSE}
source("code/helpers.R", local = knitr::knit_global())
# or sys.source("your-script.R", envir = knitr::knit_global())
```

## Data Import

We will be importing counts data from the star-salmon pipeline and our metadata 
for the project which is hosted on Box. This also ensures data is properly ordered by sample id.

```{r, import data}
counts <- read_tsv("data/star-salmon/salmon.merged.gene_counts_length_scaled.tsv")

# Import variants of interest
genes_of_interest <- read_csv("data/Prioritized_Genes_From_WGS_2024_04_25.csv")
genes_of_interest <- unique(genes_of_interest$Genes)
genes_of_interest <- genes_of_interest[!is.na(genes_of_interest)]

# Use first column (gene_id) for row names
counts <- data.frame(counts, row.names = 1)
counts$Ensembl_ID <- row.names(counts)
drop <- c("Ensembl_ID", "gene_name")
gene_info <- counts[, drop]
counts <- counts[, !(names(counts) %in% drop)] # remove both columns

# Import metadata
sample_metadata <- read_csv("data/Metadata_2024_03_14.csv")
row.names(sample_metadata) <- sample_metadata$ID

# Assuming counts is your counts dataframe and sample_metadata is your metadata dataframe
# Call the function with the appropriate column names
counts <- rename_counts_columns(counts, sample_metadata, "ID", "RNA_Samples_id")

# Check that data is ordered properly
sample_metadata <- check_order(sample_metadata = sample_metadata, counts = counts)

genes_biomart <- retrieve_gene_info(values = gene_info$Ensembl_ID, filters = "ensembl_gene_id_version")
```

## DESeq2 Analysis

```{r, start analysis}
sample_metadata$Family <- factor(sample_metadata$Family)
sample_metadata$Affected <- factor(sample_metadata$Affected)
sample_metadata$Batch <- factor(sample_metadata$Batch)
sample_metadata$Sex <- factor(sample_metadata$Sex)
sample_metadata$Ancestry <- factor(sample_metadata$Ancestry)
sample_metadata$SubCategory <- factor(sample_metadata$SubCategory)
sample_metadata$CombinedCategory <- factor(sample_metadata$CombinedCategory)

# Account for Family later but batch is accounted for
# Accounting for another factor seems to be an issue.
dds <- DESeqDataSetFromMatrix(
  countData = round(counts), colData = sample_metadata,
  design = ~ Batch + Affected
)

# Pre-filtering: Keep only rows that have at least 10 reads total
keep <- rowSums(counts(dds)) >= 10
dds <- dds[keep, ]

# Remove samples from family's with multiple samples
remove_multiple_fam <- c("8", "12", "19", "21", "23", "27")

# Save excluded/removed samples
# Code is commented out since this is for 1 time usage
# excluded_samples_df <- filter(sample_metadata,
#                        ID %in% remove_multiple_fam)
# write.csv(excluded_samples_df, "data/excluded_samples_df.csv")

# Create a logical vector to index the columns you want to keep
kept_fam_members <- !(colnames(dds) %in% remove_multiple_fam)
dds <- dds[, kept_fam_members]

# Run DESeq function
dds <- DESeq(dds)

# Normalize gene counts for differences in seq. depth/global differences
counts_norm <- counts(dds, normalized = TRUE)
```

### Data transformation and visualization

Perform count data transformation by variance stabilizing transformation (vst) on normalized counts.

```{r, vsd}
vsd <- vst(dds, blind = FALSE)
```

### Batch correction with limma

```{r, limma}
counts_vst <- assay(vsd)
write.csv(counts_vst, file = "output/counts_vst_subsetted.csv")
mm <- model.matrix(~ Batch + Affected, colData(vsd))

counts_vst_limma <- limma::removeBatchEffect(counts_vst, batch = vsd$Batch, 
                                             design = mm)

write.csv(counts_vst_limma, file = "output/counts_vst_limma_subsetted.csv")

vsd_limma <- vsd
assay(vsd_limma) <- counts_vst_limma
```

### Sample distances heatmap

```{r, sample-dist-heatmap}
sample_dists_all <- dist(t(assay(vsd_limma)))
sample_dist_matrix_all <- as.matrix(sample_dists_all)
rownames(sample_dist_matrix_all) <- paste(vsd_limma$Batch, vsd_limma$Affected, 
                                          sep = " | ")
colnames(sample_dist_matrix_all) <- paste(vsd_limma$ID, vsd_limma$Affected, 
                                          sep = " | ")
colors <- colorRampPalette(rev(brewer.pal(9, "Blues")))(255)

pheatmap(sample_dist_matrix_all, clustering_distance_rows = sample_dists_all, 
         clustering_distance_cols = sample_dists_all, col = colors)
```

### Principal Components Analysis

Our below PCA shows that there does not seem to be a batch-related effect 
occurring after using limma. However, we can see that our 3 male samples are 
grouping. Given this knowledge, removing them from downstream analyses is the best option.

```{r, principal-components-analysis-all}
pca_data_all <- plotPCA(vsd_limma, intgroup = c("Batch", "Affected", "Sex"), 
                        returnData = TRUE)
percent_var_all <- round(100 * attr(pca_data_all, "percentVar"))

ggplot(pca_data_all, aes(PC1, PC2)) +
  geom_point(aes(colour = Affected, fill = Sex, shape = Batch), size = 4) +
  scale_shape_manual(values = c(21, 22, 23)) +
  scale_fill_manual(values = c("white", "gray")) +
  scale_color_manual(values = c("blue", "red")) +
  geom_text(aes(label = name),
    data = subset(pca_data_all, PC2 < -18 | PC1 < -30),
    vjust = -1, hjust = 0.5, size = 2.5
  ) +
  xlab(paste0("PC1: ", percent_var_all[1], "% variance")) +
  ylab(paste0("PC2: ", percent_var_all[2], "% variance")) +
  coord_fixed()
```

```{r, principal-components-analysis-combocategory-all}
pca_data_all <- plotPCA(vsd_limma, intgroup = c("Affected", "Sex", 
                                                "CombinedCategory"),
                        returnData = TRUE)
percent_var_all <- round(100 * attr(pca_data_all, "percentVar"))

ggplot(pca_data_all, aes(PC1, PC2)) +
  geom_point(aes(colour = CombinedCategory, fill = Sex, shape = Affected), size = 4) +
  scale_shape_manual(values = c(21, 22, 23)) +
  scale_fill_manual(values = c("white", "gray")) +
  geom_text(aes(label = name), vjust = -1, hjust = 0.65, size = 5) +
  # geom_text_repel(aes(label = name), size = 5, vjust = -1, hjust = -0.65) +
  xlab(paste0("PC1: ", percent_var_all[1], "% variance")) +
  ylab(paste0("PC2: ", percent_var_all[2], "% variance")) +
  coord_fixed()
```

```{r, principal-components-analysis-subsetted-affected-status}
pca_data <- plotPCA(vsd_limma, intgroup = c("Sex", "Affected"), returnData = TRUE)
percent_var <- round(100 * attr(pca_data, "percentVar"))

ggplot(pca_data, aes(PC1, PC2)) +
  geom_point(aes(shape = Sex, colour = Affected), size = 5) +
  scale_color_manual(
    values = c("blue", "red"),
    name = "Affected Status",
    labels = c("Unaffected", "Affected")
  ) +
  scale_shape_manual(
    name = "Sex",
    labels = c("Male", "Female"),
    values = c(16, 17)
  ) + # Example shapes, change as needed
  geom_text_repel(aes(label = name), size = 5, vjust = -2, hjust = -0.65) +
  xlab(paste0("PC1: ", percent_var[1], "% variance")) +
  ylab(paste0("PC2: ", percent_var[2], "% variance")) +
  coord_fixed()
```

```{r, principal-components-analysis-subsetted-category}
pca_data_all <- plotPCA(vsd_limma, intgroup = c("CombinedCategory"), 
                        returnData = TRUE)
percent_var_all <- round(100 * attr(pca_data_all, "percentVar"))

ggplot(pca_data_all, aes(PC1, PC2)) +
  geom_point(aes(colour = CombinedCategory), size = 5) +
  scale_colour_manual(
    values = c(
      "#B3DE69", "#FFFFB3", "#BEBADA", "#FB8072", "#80B1D3",
      "#FCCDE5", "#8DD3C7", "#000000"
    ),
    name = "Category"
  ) +
  # geom_text(aes(label = name), vjust = -.75, hjust = -.5, size = 3) +
  # geom_text_repel(aes(label = name), size = 5, vjust = -2, hjust = -0.65) +
  geom_text_repel(aes(label = name), size = 3) +
  xlab(paste0("PC1: ", percent_var_all[1], "% variance")) +
  ylab(paste0("PC2: ", percent_var_all[2], "% variance")) +
  ggthemes::theme_clean()
```

### Heatmap of all genes, top 50  & top 100 genes

```{r, heatmap-colors, out.height="570px", out.width ="800px"}
# Specify annotation colors by columns
# Use RColorBrewer::brewer.pal(n=10, name="Set1")
category_colors <- c(
  "Autoimmune disorder" = "#B3DE69",
  "Immunodeficiency" = "#FFFFB3",
  "Channelopathy" = "#BEBADA",
  "Inflammatory disorder" = "#FB8072",
  "Metabolic disorder" = "#80B1D3",
  "Mitochondrial disorder" = "#FCCDE5",
  "Neurologic disorder" = "#D9D9D9",
  "Myasthenic disorder" = "#FDB462",
  "Neuropathy" = "#8DD3C7",
  "Unaffected" = "#001111"
)

combined_category_colors <- c(
  "Transporter-Metabolism" = "#E41A1C", # Dark green
  "Metabolism" = "#377EB8", # Dark orange
  "Metabolism-Immune" = "#4DAF4A", # Light purple
  "Immune-Transporter" = "#984EA3", # Magenta
  "Transporter" = "#FF7F00", # Light green
  "Cytoskeletal" = "#FFFF33", # Mustard yellow
  "Transporter-Immune-Metabolism" = "#F781BF", # Brown
  "Unaffected" = "#000000" # Dark grey
)


# Specify colors
ann_colors <- list(
  Batch = c(B1 = "purple", B2 = "firebrick", B3 = "yellow"),
  Affected = c(Affected = "green", Unaffected = "navy"),
  SubCategory = category_colors,
  CombinedCategory = combined_category_colors
)

ann_colors2 <- list(
  Batch = c(B1 = "purple", B2 = "firebrick", B3 = "yellow"),
  Affected = c(Affected = "green", Unaffected = "navy"),
  SubCategory = category_colors,
  CombinedCategory = combined_category_colors
)
```

This is a heatmap of the top 50 genes with the highest variance across samples

```{r, top50-heatmap-batch-combined, out.height="570px", out.width ="800px"}
top_var_genes <- head(order(-rowVars(assay(vsd_limma))), 50)
mat <- assay(vsd_limma)[top_var_genes, ]
mat <- mat - rowMeans(mat)
df <- as.data.frame(colData(vsd_limma)[, c("Batch", "Affected", "CombinedCategory")])

ensembl_to_gene <- setNames(gene_info$gene_name, gene_info$Ensembl_ID)

# Get the current row names of the matrix 'mat'
current_ensembl_ids <- rownames(mat)

# Find the corresponding gene names for the Ensembl IDs
new_row_names <- ensembl_to_gene[current_ensembl_ids]

# Set the new row names for the matrix 'mat'
rownames(mat) <- new_row_names

ComplexHeatmap::pheatmap(mat,
  annotation_col = df, annotation_colors = ann_colors2,
  fontsize = 5, angle_col = c("0")
)
```

This is a heatmap of the top 50 genes with the highest variance across all samples.

```{r, top50-heatmap-subsetted, out.height="570px", out.width ="800px"}
top_var_genes_all <- head(order(-rowVars(assay(vsd_limma))), 50)
mat_all <- assay(vsd_limma)[top_var_genes_all, ]
mat_all <- mat_all - rowMeans(mat_all)
df_all <- as.data.frame(colData(vsd_limma)[, c("Batch", "Affected", "SubCategory")])
df_affcat <- as.data.frame(colData(vsd_limma)[, c("Affected", "CombinedCategory")])

ensembl_to_gene <- setNames(gene_info$gene_name, gene_info$Ensembl_ID)

# Get the current row names of the matrix 'mat'
current_ensembl_ids_all <- rownames(mat_all)

# Find the corresponding gene names for the Ensembl IDs
new_row_names_all <- ensembl_to_gene[current_ensembl_ids_all]

# Set the new row names for the matrix 'mat'
rownames(mat_all) <- new_row_names_all

pheatmap(mat_all, annotation_col = df_all, annotation_colors = ann_colors, 
         fontsize = 5)

pheatmap(mat_all, annotation_col = df_affcat, annotation_colors = ann_colors2, 
         fontsize = 5)
```

This is a heatmap of the top 100 genes with the highest variance across
```{r, top100-heatmap-subsetted, out.width = "1200px"}
top_var_genes_100_all <- head(order(-rowVars(assay(vsd_limma))), 100)
mat_100_all <- assay(vsd_limma)[top_var_genes_100_all, ]
mat_100_all <- mat_100_all - rowMeans(mat_100_all)
df_100_all <- as.data.frame(colData(vsd_limma)[, c("Affected", "SubCategory")])

ensembl_to_gene <- setNames(gene_info$gene_name, gene_info$Ensembl_ID)

# Get the current row names of the matrix 'mat'
current_ensembl_ids_100_all <- rownames(mat_100_all)

# Find the corresponding gene names for the Ensembl IDs
new_row_names_100_all <- ensembl_to_gene[current_ensembl_ids_100_all]

# Set the new row names for the matrix 'mat'
rownames(mat_100_all) <- new_row_names_100_all

pheatmap(mat_100_all, annotation_col = df_100_all, 
         annotation_colors = ann_colors2, fontsize = 6)
```

### Comparison/Contrast of Affected_Affected_vs_Unaffected 

```{r}
res_aff_vs_unaff_sub <- results(dds, contrast = c("Affected", "Affected", "Unaffected"))
res_aff_vs_unaff_df_sub <- process_and_save_results(
  res_aff_vs_unaff_sub,
  "output/res_aff_vs_unaff_sub.csv"
)
res_aff_vs_unaff_df_sub <- arrange(res_aff_vs_unaff_df_sub, padj)
res_aff_vs_unaff_df_05_sub <- subset(res_aff_vs_unaff_df_sub, padj < 0.05)
res_aff_vs_unaff_df_1_sub <- subset(res_aff_vs_unaff_df_sub, padj < 0.1)

summary(res_aff_vs_unaff_sub)
```

```{r, top-100-genes-deseq-subsetted, out.height="570px", out.width ="800px"}
topgenes_byensemblid_sub <- head(rownames(res_aff_vs_unaff_df_sub), 100)
topgenes_aff_vs_unaff_05_sub <- assay(vsd_limma)[topgenes_byensemblid_sub, ]
topgenes_aff_vs_unaff_05_sub <- topgenes_aff_vs_unaff_05_sub - rowMeans(topgenes_aff_vs_unaff_05_sub)

ensembl_to_gene_sub <- setNames(gene_info$gene_name, gene_info$Ensembl_ID)

# Get the current row names of the matrix 'mat'
current_ensembl_ids_sub <- rownames(topgenes_aff_vs_unaff_05_sub)

# Find the corresponding gene names for the Ensembl IDs
new_row_names_sub <- ensembl_to_gene_sub[current_ensembl_ids_sub]

# Set the new row names for the matrix 'mat'
rownames(topgenes_aff_vs_unaff_05_sub) <- new_row_names_sub

topgenes_aff_vs_unaff_05_sub <- topgenes_aff_vs_unaff_05_sub[order(row.names(topgenes_aff_vs_unaff_05_sub)), ]

df_sub <- as.data.frame(colData(vsd_limma)[, c("Affected", "CombinedCategory")])
pheatmap(topgenes_aff_vs_unaff_05_sub, annotation_col = df_sub, annotation_colors = ann_colors2, fontsize = 5, angle_col = c("0"))
```

```{r, top-100-genes-deseq-subsetted-subcategory, out.height="570px", out.width ="800px"}
topgenes_byensemblid_sub <- head(rownames(res_aff_vs_unaff_df_sub), 100)
topgenes_aff_vs_unaff_05_sub <- assay(vsd_limma)[topgenes_byensemblid_sub, ]
topgenes_aff_vs_unaff_05_sub <- topgenes_aff_vs_unaff_05_sub - rowMeans(topgenes_aff_vs_unaff_05_sub)

ensembl_to_gene_sub <- setNames(gene_info$gene_name, gene_info$Ensembl_ID)

# Get the current row names of the matrix 'mat'
current_ensembl_ids_sub <- rownames(topgenes_aff_vs_unaff_05_sub)

# Find the corresponding gene names for the Ensembl IDs
new_row_names_sub <- ensembl_to_gene_sub[current_ensembl_ids_sub]

# Set the new row names for the matrix 'mat'
rownames(topgenes_aff_vs_unaff_05_sub) <- new_row_names_sub

topgenes_aff_vs_unaff_05_sub <- topgenes_aff_vs_unaff_05_sub[order(row.names(topgenes_aff_vs_unaff_05_sub)), ]

df_sub <- as.data.frame(colData(vsd_limma)[, c("Affected", "SubCategory", "CombinedCategory")])
pheatmap(topgenes_aff_vs_unaff_05_sub, annotation_col = df_sub, annotation_colors = ann_colors2, fontsize = 5, angle_col = 0)
```


```{r, top-50-genes-deseq-subsetted, out.height="570px", out.width ="800px"}
topgenes_byensemblid_sub <- head(rownames(res_aff_vs_unaff_df_sub), 50)
topgenes_aff_vs_unaff_05_sub <- assay(vsd_limma)[topgenes_byensemblid_sub, ]
topgenes_aff_vs_unaff_05_sub <- topgenes_aff_vs_unaff_05_sub - rowMeans(topgenes_aff_vs_unaff_05_sub)

ensembl_to_gene_sub <- setNames(gene_info$gene_name, gene_info$Ensembl_ID)

# Get the current row names of the matrix 'mat'
current_ensembl_ids_sub <- rownames(topgenes_aff_vs_unaff_05_sub)

# Find the corresponding gene names for the Ensembl IDs
new_row_names_sub <- ensembl_to_gene_sub[current_ensembl_ids_sub]

# Set the new row names for the matrix 'mat'
rownames(topgenes_aff_vs_unaff_05_sub) <- new_row_names_sub

topgenes_aff_vs_unaff_05_sub <- topgenes_aff_vs_unaff_05_sub[order(row.names(topgenes_aff_vs_unaff_05_sub)), ]

df_sub <- df <- colData(vsd_limma) %>%
  as.data.frame() %>%
  dplyr::select(CombinedCategory)
pheatmap(topgenes_aff_vs_unaff_05_sub, annotation_col = df_sub, annotation_colors = ann_colors2, fontsize = 5, angle_col = 0)
```

```{r, top-50-genes-deseq-subsetted-subcategory, out.height="570px", out.width ="800px"}
topgenes_byensemblid_sub <- head(rownames(res_aff_vs_unaff_df_sub), 50)
topgenes_aff_vs_unaff_05_sub <- assay(vsd_limma)[topgenes_byensemblid_sub, ]
topgenes_aff_vs_unaff_05_sub <- topgenes_aff_vs_unaff_05_sub - rowMeans(topgenes_aff_vs_unaff_05_sub)

ensembl_to_gene_sub <- setNames(gene_info$gene_name, gene_info$Ensembl_ID)

# Get the current row names of the matrix 'mat'
current_ensembl_ids_sub <- rownames(topgenes_aff_vs_unaff_05_sub)

# Find the corresponding gene names for the Ensembl IDs
new_row_names_sub <- ensembl_to_gene_sub[current_ensembl_ids_sub]

# Set the new row names for the matrix 'mat'
rownames(topgenes_aff_vs_unaff_05_sub) <- new_row_names_sub

topgenes_aff_vs_unaff_05_sub <- topgenes_aff_vs_unaff_05_sub[order(row.names(topgenes_aff_vs_unaff_05_sub)), ]

df_sub <- df <- colData(vsd_limma) %>%
  as.data.frame() %>%
  dplyr::select(Affected, SubCategory, CombinedCategory)
pheatmap(topgenes_aff_vs_unaff_05_sub,
  annotation_col = df_sub,
  annotation_colors = ann_colors2, fontsize = 5, angle_col = 0
)
```


```{r, significant-genes-deseq-subsetted, out.height="570px", out.width ="800px"}
topgenes_byensemblid_sub <- head(rownames(res_aff_vs_unaff_df_sub), 55)
topgenes_aff_vs_unaff_05_sub <- assay(vsd_limma)[topgenes_byensemblid_sub, ]
topgenes_aff_vs_unaff_05_sub <- topgenes_aff_vs_unaff_05_sub - rowMeans(topgenes_aff_vs_unaff_05_sub)

ensembl_to_gene_sub <- setNames(gene_info$gene_name, gene_info$Ensembl_ID)

# Get the current row names of the matrix 'mat'
current_ensembl_ids_sub <- rownames(topgenes_aff_vs_unaff_05_sub)

# Find the corresponding gene names for the Ensembl IDs
new_row_names_sub <- ensembl_to_gene_sub[current_ensembl_ids_sub]

# Set the new row names for the matrix 'mat'
rownames(topgenes_aff_vs_unaff_05_sub) <- new_row_names_sub

topgenes_aff_vs_unaff_05_sub <- topgenes_aff_vs_unaff_05_sub[order(row.names(topgenes_aff_vs_unaff_05_sub)), ]

df_sub <- df <- colData(vsd_limma) %>%
  as.data.frame() %>%
  dplyr::select(CombinedCategory)
pheatmap(topgenes_aff_vs_unaff_05_sub, annotation_col = df_sub, annotation_colors = ann_colors2, fontsize = 5, angle_col = 0)
```


```{r, significant-genes-deseq-subsetted-subcategory, out.height="570px", out.width ="800px"}
topgenes_byensemblid_sub <- head(rownames(res_aff_vs_unaff_df_sub), 55)
topgenes_aff_vs_unaff_05_sub <- assay(vsd_limma)[topgenes_byensemblid_sub, ]
topgenes_aff_vs_unaff_05_sub <- topgenes_aff_vs_unaff_05_sub - rowMeans(topgenes_aff_vs_unaff_05_sub)

ensembl_to_gene_sub <- setNames(gene_info$gene_name, gene_info$Ensembl_ID)

# Get the current row names of the matrix 'mat'
current_ensembl_ids_sub <- rownames(topgenes_aff_vs_unaff_05_sub)

# Find the corresponding gene names for the Ensembl IDs
new_row_names_sub <- ensembl_to_gene_sub[current_ensembl_ids_sub]

# Set the new row names for the matrix 'mat'
rownames(topgenes_aff_vs_unaff_05_sub) <- new_row_names_sub

topgenes_aff_vs_unaff_05_sub <- topgenes_aff_vs_unaff_05_sub[order(row.names(topgenes_aff_vs_unaff_05_sub)), ]

df_sub <- df <- colData(vsd_limma) %>%
  as.data.frame() %>%
  dplyr::select(Affected, SubCategory, CombinedCategory)
pheatmap(topgenes_aff_vs_unaff_05_sub, annotation_col = df_sub, annotation_colors = ann_colors2, fontsize = 5, angle_col = 0)
```

```{r, top-75-genes-deseq-subsetted, out.height="570px", out.width ="800px"}
topgenes_byensemblid_sub <- head(rownames(res_aff_vs_unaff_df_sub), 75)
topgenes_aff_vs_unaff_05_sub <- assay(vsd_limma)[topgenes_byensemblid_sub, ]
topgenes_aff_vs_unaff_05_sub <- topgenes_aff_vs_unaff_05_sub - rowMeans(topgenes_aff_vs_unaff_05_sub)

ensembl_to_gene_sub <- setNames(gene_info$gene_name, gene_info$Ensembl_ID)

# Get the current row names of the matrix 'mat'
current_ensembl_ids_sub <- rownames(topgenes_aff_vs_unaff_05_sub)

# Find the corresponding gene names for the Ensembl IDs
new_row_names_sub <- ensembl_to_gene_sub[current_ensembl_ids_sub]

# Set the new row names for the matrix 'mat'
rownames(topgenes_aff_vs_unaff_05_sub) <- new_row_names_sub

topgenes_aff_vs_unaff_05_sub <- topgenes_aff_vs_unaff_05_sub[order(row.names(topgenes_aff_vs_unaff_05_sub)), ]

df_sub <- as.data.frame(colData(vsd_limma)[, c("Affected", "CombinedCategory")])
pheatmap(topgenes_aff_vs_unaff_05_sub,
  annotation_col = df_sub,
  annotation_colors = ann_colors2, fontsize = 4, angle_col = 0
)
```

```{r, top-75-genes-deseq-subsetted-subcategory, out.height="570px", out.width ="800px"}
topgenes_byensemblid_sub <- head(rownames(res_aff_vs_unaff_df_sub), 75)
topgenes_aff_vs_unaff_05_sub <- assay(vsd_limma)[topgenes_byensemblid_sub, ]
topgenes_aff_vs_unaff_05_sub <- topgenes_aff_vs_unaff_05_sub - rowMeans(topgenes_aff_vs_unaff_05_sub)

ensembl_to_gene_sub <- setNames(gene_info$gene_name, gene_info$Ensembl_ID)

# Get the current row names of the matrix 'mat'
current_ensembl_ids_sub <- rownames(topgenes_aff_vs_unaff_05_sub)

# Find the corresponding gene names for the Ensembl IDs
new_row_names_sub <- ensembl_to_gene_sub[current_ensembl_ids_sub]

# Set the new row names for the matrix 'mat'
rownames(topgenes_aff_vs_unaff_05_sub) <- new_row_names_sub

topgenes_aff_vs_unaff_05_sub <- topgenes_aff_vs_unaff_05_sub[order(row.names(topgenes_aff_vs_unaff_05_sub)), ]

df_sub <- as.data.frame(colData(vsd_limma)[, c("Affected", "SubCategory", "CombinedCategory")])
pheatmap(topgenes_aff_vs_unaff_05_sub, annotation_col = df_sub, annotation_colors = ann_colors2, fontsize = 4, angle_col = 0)
```

```{r, top-200-genes-deseq-subsetted, out.height="570px", out.width ="800px"}
topgenes_byensemblid_sub <- head(rownames(res_aff_vs_unaff_df_sub), 200)
topgenes_aff_vs_unaff_05_sub <- assay(vsd_limma)[topgenes_byensemblid_sub, ]
topgenes_aff_vs_unaff_05_sub <- topgenes_aff_vs_unaff_05_sub - rowMeans(topgenes_aff_vs_unaff_05_sub)

ensembl_to_gene_sub <- setNames(gene_info$gene_name, gene_info$Ensembl_ID)

# Get the current row names of the matrix 'mat'
current_ensembl_ids_sub <- rownames(topgenes_aff_vs_unaff_05_sub)

# Find the corresponding gene names for the Ensembl IDs
new_row_names_sub <- ensembl_to_gene_sub[current_ensembl_ids_sub]

# Set the new row names for the matrix 'mat'
rownames(topgenes_aff_vs_unaff_05_sub) <- new_row_names_sub

topgenes_aff_vs_unaff_05_sub <- topgenes_aff_vs_unaff_05_sub[order(row.names(topgenes_aff_vs_unaff_05_sub)), ]

df_sub <- as.data.frame(colData(vsd_limma)[, c("Affected", "CombinedCategory")])
pheatmap(topgenes_aff_vs_unaff_05_sub, annotation_col = df_sub, annotation_colors = ann_colors2, fontsize = 4, angle_col = 0)
```

```{r, top-200-genes-deseq-subsetted-subcategory, out.height="570px", out.width ="800px"}
topgenes_byensemblid_sub <- head(rownames(res_aff_vs_unaff_df_sub), 200)
topgenes_aff_vs_unaff_05_sub <- assay(vsd_limma)[topgenes_byensemblid_sub, ]
topgenes_aff_vs_unaff_05_sub <- topgenes_aff_vs_unaff_05_sub - rowMeans(topgenes_aff_vs_unaff_05_sub)

ensembl_to_gene_sub <- setNames(gene_info$gene_name, gene_info$Ensembl_ID)

# Get the current row names of the matrix 'mat'
current_ensembl_ids_sub <- rownames(topgenes_aff_vs_unaff_05_sub)

# Find the corresponding gene names for the Ensembl IDs
new_row_names_sub <- ensembl_to_gene_sub[current_ensembl_ids_sub]

# Set the new row names for the matrix 'mat'
rownames(topgenes_aff_vs_unaff_05_sub) <- new_row_names_sub

topgenes_aff_vs_unaff_05_sub <- topgenes_aff_vs_unaff_05_sub[order(row.names(topgenes_aff_vs_unaff_05_sub)), ]

df_sub <- as.data.frame(colData(vsd_limma)[, c("Affected", "SubCategory", "CombinedCategory")])
pheatmap(topgenes_aff_vs_unaff_05_sub, annotation_col = df_sub, annotation_colors = ann_colors2, fontsize = 2.25, angle_col = 0)
```

```{r, top-125-genes-deseq-all-subsetted, out.height="570px", out.width ="800px"}
topgenes_byensemblid_sub <- head(rownames(res_aff_vs_unaff_df_sub), 125)
topgenes_aff_vs_unaff_05_sub <- assay(vsd_limma)[topgenes_byensemblid_sub, ]
topgenes_aff_vs_unaff_05_sub <- topgenes_aff_vs_unaff_05_sub - rowMeans(topgenes_aff_vs_unaff_05_sub)

ensembl_to_gene_sub <- setNames(gene_info$gene_name, gene_info$Ensembl_ID)

# Get the current row names of the matrix 'mat'
current_ensembl_ids_sub <- rownames(topgenes_aff_vs_unaff_05_sub)

# Find the corresponding gene names for the Ensembl IDs
new_row_names_sub <- ensembl_to_gene_sub[current_ensembl_ids_sub]

# Set the new row names for the matrix 'mat'
rownames(topgenes_aff_vs_unaff_05_sub) <- new_row_names_sub

# topgenes_aff_vs_unaff_05_sub <- topgenes_aff_vs_unaff_05_sub[order(row.names(topgenes_aff_vs_unaff_05_sub)), ]

df_sub <- as.data.frame(colData(vsd_limma)[, c("Affected", "CombinedCategory")])
pheatmap(topgenes_aff_vs_unaff_05_sub, annotation_col = df_sub, annotation_colors = ann_colors2, fontsize = 3.75, angle_col = 0)
```

```{r, top-125-genes-deseq-all-subsetted-subcategory, out.height="570px", out.width ="800px"}
topgenes_byensemblid_sub <- head(rownames(res_aff_vs_unaff_df_sub), 125)
topgenes_aff_vs_unaff_05_sub <- assay(vsd_limma)[topgenes_byensemblid_sub, ]
topgenes_aff_vs_unaff_05_sub <- topgenes_aff_vs_unaff_05_sub - rowMeans(topgenes_aff_vs_unaff_05_sub)

ensembl_to_gene_sub <- setNames(gene_info$gene_name, gene_info$Ensembl_ID)

# Get the current row names of the matrix 'mat'
current_ensembl_ids_sub <- rownames(topgenes_aff_vs_unaff_05_sub)

# Find the corresponding gene names for the Ensembl IDs
new_row_names_sub <- ensembl_to_gene_sub[current_ensembl_ids_sub]

# Set the new row names for the matrix 'mat'
rownames(topgenes_aff_vs_unaff_05_sub) <- new_row_names_sub

# topgenes_aff_vs_unaff_05_sub <- topgenes_aff_vs_unaff_05_sub[order(row.names(topgenes_aff_vs_unaff_05_sub)), ]

df_sub <- as.data.frame(colData(vsd_limma)[, c("Affected", "SubCategory", "CombinedCategory")])
pheatmap(topgenes_aff_vs_unaff_05_sub, annotation_col = df_sub, annotation_colors = ann_colors2, fontsize = 3.75, angle_col = 0)
```


```{r, top-150-genes-deseq-all-subsetted, out.height="570px", out.width ="800px"}
topgenes_byensemblid_sub <- head(rownames(res_aff_vs_unaff_df_sub), 150)
topgenes_aff_vs_unaff_05_sub <- assay(vsd_limma)[topgenes_byensemblid_sub, ]
topgenes_aff_vs_unaff_05_sub <- topgenes_aff_vs_unaff_05_sub - rowMeans(topgenes_aff_vs_unaff_05_sub)

ensembl_to_gene_sub <- setNames(gene_info$gene_name, gene_info$Ensembl_ID)

# Get the current row names of the matrix 'mat'
current_ensembl_ids_sub <- rownames(topgenes_aff_vs_unaff_05_sub)

# Find the corresponding gene names for the Ensembl IDs
new_row_names_sub <- ensembl_to_gene_sub[current_ensembl_ids_sub]

# Set the new row names for the matrix 'mat'
rownames(topgenes_aff_vs_unaff_05_sub) <- new_row_names_sub

# topgenes_aff_vs_unaff_05_sub <- topgenes_aff_vs_unaff_05_sub[order(row.names(topgenes_aff_vs_unaff_05_sub)), ]

df_sub <- as.data.frame(colData(vsd_limma)[, c("Affected", "CombinedCategory")])
pheatmap(topgenes_aff_vs_unaff_05_sub, annotation_col = df_sub, annotation_colors = ann_colors2, fontsize = 3.5, angle_col = 0)
```

```{r, top-150-genes-deseq-subsetted-subcategory, out.height="570px", out.width ="800px"}
topgenes_byensemblid_sub <- head(rownames(res_aff_vs_unaff_df_sub), 150)
topgenes_aff_vs_unaff_05_sub <- assay(vsd_limma)[topgenes_byensemblid_sub, ]
topgenes_aff_vs_unaff_05_sub <- topgenes_aff_vs_unaff_05_sub - rowMeans(topgenes_aff_vs_unaff_05_sub)

ensembl_to_gene_sub <- setNames(gene_info$gene_name, gene_info$Ensembl_ID)

# Get the current row names of the matrix 'mat'
current_ensembl_ids_sub <- rownames(topgenes_aff_vs_unaff_05_sub)

# Find the corresponding gene names for the Ensembl IDs
new_row_names_sub <- ensembl_to_gene_sub[current_ensembl_ids_sub]

# Set the new row names for the matrix 'mat'
rownames(topgenes_aff_vs_unaff_05_sub) <- new_row_names_sub

# topgenes_aff_vs_unaff_05_sub <- topgenes_aff_vs_unaff_05_sub[order(row.names(topgenes_aff_vs_unaff_05_sub)), ]

df_sub <- as.data.frame(colData(vsd_limma)[, c("Affected", "SubCategory", "CombinedCategory")])
pheatmap(topgenes_aff_vs_unaff_05_sub, annotation_col = df_sub, annotation_colors = ann_colors2, fontsize = 3.5, angle_col = 0)
```


```{r, all-genes-list-subsetted}
gb_df <- genes_biomart[, c(1, ncol(genes_biomart))]
res_aff_vs_unaff_df_sub_genename <- res_aff_vs_unaff_df_sub
res_aff_vs_unaff_df_sub_genename$Ensembl_ID <- row.names(res_aff_vs_unaff_df_sub)
res_aff_vs_unaff_df_sub_genename <- merge(x = res_aff_vs_unaff_df_sub_genename, y = gene_info, by.x = "Ensembl_ID", by.y = "Ensembl_ID", all.x = TRUE)
res_aff_vs_unaff_df_sub_genename <- res_aff_vs_unaff_df_sub_genename[, c(dim(res_aff_vs_unaff_df_sub_genename)[2], 1:dim(res_aff_vs_unaff_df_sub_genename)[2] - 1)]
res_aff_vs_unaff_df_sub_genename <- res_aff_vs_unaff_df_sub_genename[order(res_aff_vs_unaff_df_sub_genename[, "padj"]), ]
write.csv(res_aff_vs_unaff_df_sub_genename, file = "output/res_aff_vs_unaff_df_sub_genename.csv")
```

```{r, top50-with-genename-padj}
res_aff_vs_unaff_df_genename_05 <- subset(res_aff_vs_unaff_df_sub_genename, padj < 0.05)
res_aff_vs_unaff_df_genename_05 <- res_aff_vs_unaff_df_genename_05[order(res_aff_vs_unaff_df_genename_05$padj), ]
write.csv(res_aff_vs_unaff_df_genename_05, file = "output/res_aff_vs_unaff_df_sub_genename_05.csv")

res_aff_vs_unaff_df_genename_1 <- subset(res_aff_vs_unaff_df_sub_genename, padj < 0.1)
res_aff_vs_unaff_df_genename_1 <- res_aff_vs_unaff_df_genename_1[order(res_aff_vs_unaff_df_genename_1$padj), ]
write.csv(res_aff_vs_unaff_df_genename_1, file = "output/res_aff_vs_unaff_df_sub_genename_padj1.csv")
```

Below is a table of information about the top genes.

```{r, padj-gene-info-subsetted}
library(mygene)

genes <- res_aff_vs_unaff_df_genename_05$gene_name
my_gene_data <- queryMany(genes, scopes = "symbol", fields = c("symbol", "name", "summary", species = "human"))
my_gene_data_unique <- as.data.frame(my_gene_data) %>% dplyr::distinct(query, .keep_all = TRUE)
# paged_table(my_gene_data, options = list(rows.print = 15))
```

```{r, maplot-subsetted-samples}
filtered_gene_names <- res_aff_vs_unaff_df_sub_genename$gene_name[!grepl("^ENS", res_aff_vs_unaff_df_sub_genename$gene_name)]
# Select specific genes to show
# set top = 0, then specify genes using label.select argument
maplot <- ggmaplot(res_aff_vs_unaff_df_sub,
  main = "Affected vs Unaffected MA Plot",
  fdr = .1, fc = 1, size = 0.4, # Same used for others.
  genenames = as.vector(res_aff_vs_unaff_df_sub_genename$gene_name),
  ggtheme = ggplot2::theme_minimal(),
  legend = "top", top = 30,
  label.select = c("KCNQ5"),
  font.label = c("bold", 6), label.rectangle = TRUE,
  font.legend = "bold", font.main = "bold"
)

maplot

significant_data <- maplot$data %>%
  filter(grepl("Up|Down", sig)) %>%
  mutate(direction = ifelse(grepl("Up", sig), "Up", "Down")) %>%
  dplyr::select(-sig) # This removes the 'sig' column

# Combine DataFrames based on matching 'query' in my_gene_data_unique to 'gene' in significant_data
combined_data <- inner_join(my_gene_data_unique, significant_data, by = c("query" = "name"))

combined_data <- combined_data %>%
  dplyr::select(-notfound, -X_id, -X_score) %>%
  rename(gene = query)

paged_table(as.data.frame(significant_data), options = list(rows.print = 30))
# Save significant genes
write.csv(significant_data, file = "output/res_aff_vs_unaff_significant_subsetted_samples.csv", row.names = FALSE)

# Save significant genes
write.csv(combined_data, file = "output/res_aff_vs_unaff_significant_subsetted_mygene.csv", row.names = FALSE)
```

## Expression of Candidate Genes

Below is a table of expression of the genes identified during our WGS analysis.

```{r, genes-of-interest-table-subsetted}
# Subset gene_info using genes of interest
subset_gene_info <- gene_info[gene_info$gene_name %in% genes_of_interest, ]

filtered_by_interest <- filter(res_aff_vs_unaff_df_sub_genename, Ensembl_ID %in% subset_gene_info$Ensembl_ID)

# Filtering the dataframe by row names
filtered_counts <- counts[row.names(counts) %in% subset_gene_info$Ensembl_ID, ]

# Match and update row names
matching_indices <- match(rownames(filtered_counts), subset_gene_info$Ensembl_ID)

# Update row names based on matching_column values from metadata
rownames(filtered_counts) <- subset_gene_info$gene_name[matching_indices]

missing_genes <- setdiff(subset_gene_info$Ensembl_ID, filtered_by_interest$Ensembl_ID)

corresponding_gene_names <- subset_gene_info$gene_name[subset_gene_info$Ensembl_ID %in% missing_genes]

missing_gene_counts <- counts[row.names(counts) %in% missing_genes, ]

paged_table(filtered_by_interest, options = list(rows.print = 15))

paged_table(filtered_counts, options = list(rows.print = 15))
```


Below is a heatmap highlighting the expression of our genes of interest across batch, disease group, and affected status.

```{r, heatmap-genes-interest-subsetted, out.width = "1200px"}
mat3 <- assay(vsd_limma)[filtered_by_interest$Ensembl_ID, ]
rownames(mat3) <- gene_info$gene_name[match(filtered_by_interest$Ensembl_ID, gene_info$Ensembl_ID)]
df_sub <- as.data.frame(colData(vsd_limma)[, c("Affected", "CombinedCategory")])
pheatmap(mat3, annotation_col = df_sub, annotation_colors = ann_colors2, fontsize = 4, angle_col = 0)
```

## Enrichment analysis

```{r, gostplot}
# res <- gost(query = significant_data$name, organism = "hsapiens", significant = FALSE) # should significant be true or false?
# 
# x <- gostplot(res, capped = FALSE, interactive = FALSE)
# x + geom_label_repel(aes(label = "term_id"),
#   box.padding   = 0.35,
#   point.padding = 0.5,
#   segment.color = "grey50"
# )
```

```{r, enrichgo-cc}
# Your existing code for obtaining geneList with Ensembl IDs
rownames(res_aff_vs_unaff_df_sub) <- gsub("\\..*", "", rownames(res_aff_vs_unaff_df_sub))
res_aff_vs_unaff_df_sub <- res_aff_vs_unaff_df_sub %>% dplyr::distinct()
geneList <- res_aff_vs_unaff_df_sub$log2FoldChange
geneList2 <- res_aff_vs_unaff_df_sub$padj
geneList3 <- res_aff_vs_unaff_df_sub$padj
names(geneList3) <- rownames(res_aff_vs_unaff_df_sub)
names(geneList2) <- rownames(res_aff_vs_unaff_df_sub)
names(geneList) <- rownames(res_aff_vs_unaff_df_sub)

# Initialize BioMart
ensembl <- useMart("ensembl", dataset = "hsapiens_gene_ensembl")

# Convert all Ensembl IDs in geneList to Entrez IDs
attributes <- c("ensembl_gene_id", "entrezgene_id")
filters <- "ensembl_gene_id"
results <- getBM(attributes = attributes, filters = filters, values = names(geneList), mart = ensembl)
results3 <- getBM(attributes = attributes, filters = filters, values = names(geneList2), mart = ensembl)

# Convert all Ensembl IDs in geneList to Entrez IDs
# This is for genes of interest
attributes2 <- c("hgnc_symbol", "ensembl_gene_id", "entrezgene_id")
filters2 <- "hgnc_symbol"
results2 <- getBM(attributes = attributes2, filters = filters2, values = genes_of_interest, mart = ensembl)

# Remove rows with NA or empty Entrez IDs
results <- results[!is.na(results$entrezgene_id) & results$entrezgene_id != "", ]
results3 <- results3[!is.na(results3$entrezgene_id) & results3$entrezgene_id != "", ]

results2 <- results2[!is.na(results2$entrezgene_id) & results2$entrezgene_id != "", ]

# Create a new geneList with Entrez IDs
geneList_entrez <- geneList[results$ensembl_gene_id]
names(geneList_entrez) <- results$entrezgene_id

# Create a new geneList with Entrez IDs
geneList_entrez2 <- geneList2[results3$ensembl_gene_id]
names(geneList_entrez2) <- results3$entrezgene_id

# Create a new geneList with Entrez IDs
geneList_entrez3 <- geneList3[results2$ensembl_gene_id]
names(geneList_entrez3) <- results2$entrezgene_id

# Filter genes with abs(log2FoldChange) > 2
gene_entrez <- names(geneList_entrez)[abs(geneList_entrez) > 2]

# Filter genes with padj < .05
gene_entrez2 <- names(geneList_entrez2)[!is.na(geneList_entrez2) & geneList_entrez2 < .05]

# Run enrichGO with Entrez IDs
ego_cc <- enrichGO(
  gene = gene_entrez,
  universe = names(geneList_entrez),
  OrgDb = org.Hs.eg.db,
  ont = "CC",
  pAdjustMethod = "BH",
  pvalueCutoff = 0.01,
  qvalueCutoff = 0.05,
  readable = TRUE
)

# Plot results
dotplot(ego_cc, showCategory = 10, title = "Overrepresented Cellular Components") + ggthemes::theme_clean()
```

```{r, enrichgo-bp}
ego_bp <- enrichGO(
  gene = gene_entrez,
  universe = names(geneList_entrez),
  OrgDb = org.Hs.eg.db,
  ont = "BP",
  pAdjustMethod = "BH",
  pvalueCutoff = 0.01,
  qvalueCutoff = 0.05,
  readable = TRUE
)

# Plot results
dotplot(ego_bp, showCategory = 10, title = "Overrepresented Biological Processes") + ggthemes::theme_clean()
```

```{r, enrichgo-mf}
ego_mf <- enrichGO(
  gene = gene_entrez,
  universe = names(geneList_entrez),
  OrgDb = org.Hs.eg.db,
  ont = "MF",
  pAdjustMethod = "BH",
  pvalueCutoff = 0.01,
  qvalueCutoff = 0.05,
  readable = TRUE
)

# Plot results
dotplot(ego_mf, showCategory = 10, title = "Overrepresented Molecular Functions") + ggthemes::theme_clean()
```


```{r, dose-plot}
library(DOSE)
gl <- geneList_entrez[abs(geneList_entrez) > 2]
gl <- na.omit(gl)
gl_sorted <- sort(gl, decreasing = TRUE)

gse <- gseGO(
  geneList = gl_sorted,
  OrgDb = org.Hs.eg.db,
  ont = "ALL",
  minGSSize = 3,
  maxGSSize = 500,
  pvalueCutoff = 0.05,
  verbose = FALSE
)

dotplot(gse, showCategory = 10, split = ".sign", title = "GO Gene Set Enrichment Analysis") + facet_grid(. ~ .sign)
```


```{r, kegg-plot}
kk2 <- gseKEGG(
  geneList = gl_sorted,
  organism = "hsa", # humans = hsa for KEGG
  nPerm = 100000,
  minGSSize = 3,
  maxGSSize = 800,
  pvalueCutoff = 0.05,
  pAdjustMethod = "none",
  keyType = "ncbi-geneid"
)

dotplot(kk2, showCategory = 10, title = "Enriched KEGG Pathways", split = ".sign") + facet_grid(. ~ .sign)
```

```{r, reactome-plot}
y <- enrichPathway(gene_entrez, pvalueCutoff = 0.05)

dotplot(y, showCategory = 10, title = "Reactome Pathway Over-representation Analysis", font.size = 10)
```


```{r, stringdb-network-significant}
library(STRINGdb)
string_db <- STRINGdb$new(version = "11.5", species = 9606, score_threshold = 100, network_type = "full", input_directory = "")

mapped_degs <- string_db$map(significant_data, "name", removeUnmappedRows = TRUE)
hits <- mapped_degs$STRING_id[1:29]

mapped_degs_dec <- string_db$add_diff_exp_color(mapped_degs, logFcColStr = "lfc")

payload_id <- string_db$post_payload(mapped_degs_dec$STRING_id, colors = mapped_degs_dec$color)
string_db$plot_network(hits, payload_id = payload_id)
```

```{r, stringdb-network-combined}
library(STRINGdb)
string_db <- STRINGdb$new(version = "11.5", species = 9606, score_threshold = 200, network_type = "full", input_directory = "")

significant_data_copy <- significant_data %>% 
  rename(gene_name = name, log2FoldChange = lfc, baseMean = mean)

# Ensure both dataframes have the same columns (optional step to demonstrate alignment)
# This step is not necessary if you are sure both dataframes are already aligned
significant_data_copy <- significant_data_copy[c("gene_name", "baseMean", "log2FoldChange")]
filtered_by_interest2 <- filtered_by_interest[c("gene_name", "baseMean", "log2FoldChange")]

combined_df <- dplyr::bind_rows(significant_data_copy, filtered_by_interest2)

combined_df_mapped <- string_db$map(combined_df, "gene_name", removeUnmappedRows = TRUE)

combined_hits <- combined_df_mapped$STRING_id

string_db$plot_network(combined_hits)
combined_enrichment <- string_db$get_enrichment(combined_hits)

enrich_file_name <- "output/stringdb_combined_enrichment.csv"
write.csv(combined_enrichment, enrich_file_name, row.names = FALSE)
```


```{r, enrichr-analysis}
library(enrichR)
listEnrichrSites()
setEnrichrSite("Enrichr") # Human genes
dbs <- listEnrichrDbs()
#query_dbs <- c("GO_Molecular_Function_2015", "GO_Cellular_Component_2015", "GO_Biological_Process_2015")
enriched <- enrichr(genes = significant_data$name, databases = dbs$libraryName)

db_names <- names(enriched)

for (db_name in db_names) {
  df <- as.data.frame(enriched[[db_name]]) # Access the data frame by name
  if (is.data.frame(df)) { # Ensure it is a data frame
    file_name <- paste0("output/enrichr-dbs/", db_name, ".csv")
    write.csv(df, file_name, row.names = FALSE) # Write to CSV without row names
  }
}
```

```{r, compare-clusters-enrich-go}
cluster <- list("Significant Genes" = gene_entrez2, "Genes of Interest" = as.character(results2$entrezgene_id))
ck <- compareCluster(geneCluster = cluster, fun = "enrichGO", OrgDb='org.Hs.eg.db')
ck <- setReadable(ck, OrgDb = org.Hs.eg.db, keyType="ENTREZID")
dotplot(ck)
```

```{r, compare-clusters-enrich-kegg}
xx <- compareCluster(cluster, fun="enrichKEGG",
                     organism="hsa", pvalueCutoff=0.05)
dotplot(xx)
```

```{r, compare-clusters-enrich-pathway}
xx <- compareCluster(cluster, fun="enrichPathway", pvalueCutoff=0.05)
dotplot(xx)
```

## Plot Counts

```{r, generate-plot-counts-genes-interest-subsetted}
# For loop for plot counts
for (ensembl_id in filtered_by_interest$Ensembl_ID) {
  d <- plotCounts(dds, gene = ensembl_id, intgroup = "Affected", returnData = TRUE)
  gene_name <- filtered_by_interest$gene_name[filtered_by_interest$Ensembl_ID == ensembl_id]
  ggplot_box <- ggboxplot(d,
    x = "Affected", y = "count", add = "jitter",
    color = "Affected", palette = c("red", "navy"),
    title = gene_name
  ) +
    geom_text_repel(aes(label = rownames(d))) +
    scale_color_manual(
      name = "Affected Status",
      labels = c("Unaffected", "Affected"),
      values = c("red", "navy")
    )
  print(ggplot_box) # Ensure each plot is printed during the loop
  ggsave(
    filename = paste0(gene_name, "-subsetted-plot-counts.png"),
    path = "output/batch-correction-limma/plot-counts",
    plot = ggplot_box, dpi = 450
  )
}
```

## Combined Heatmap of DEGs and Patient Genes

```{r, combined-heatmap-genes-interest-subsetted, out.width = "1200px"}
mat3 <- assay(vsd_limma)[filtered_by_interest$Ensembl_ID, ]
rownames(mat3) <- gene_info$gene_name[match(filtered_by_interest$Ensembl_ID, gene_info$Ensembl_ID)]
df_sub <- as.data.frame(colData(vsd_limma)[, c("Affected", "CombinedCategory")])
pheatmap(mat3, annotation_col = df_sub, annotation_colors = ann_colors2, fontsize = 4, angle_col = 0)
```

### Save data

```{r, save-data}
save.image(file = "output/subsetted-condition-analysis.RData")
```

