---
title: "Differential Gene Expression (DGE) Analysis"
output:
  workflowr::wflow_html:
    toc: true
    toc_depth: 3
    toc_float: true
    code_folding: hide
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
```

# DGE Additional Analysis by Tier

All libraries from the prior analysis should be loaded.

```{r, load libraries}
library(tidyverse) # Available via CRAN
library(DESeq2) # Available via Bioconductor
library(RColorBrewer) # Available via CRAN
library(pheatmap) # Available via CRAN
library(genefilter) # Available via Bioconductor
library(limma) # Available via Bioconductor
library(gprofiler2) # Available via CRAN
library(biomaRt) # Available via Bioconductor
library(plotly) # Available via CRAN
library(ggpubr)
library(rmarkdown)
```

```{r, include=FALSE}
source("code/helpers.R", local = knitr::knit_global())
# or sys.source("your-script.R", envir = knitr::knit_global())
```

## Data Import

We will be importing counts data from the star-salmon pipeline and our metadata 
for the project which is hosted on Box. This also ensures data is properly ordered by sample id.

```{r, import data}
counts <- read_tsv("data/star-salmon/salmon.merged.gene_counts_length_scaled.tsv")

# Import variants of interest
vars_of_interest <- read_csv("data/RNAseq Variants of Interest List_V2.csv")
genes_of_interest <- unique(vars_of_interest$Gene)
genes_of_interest <- genes_of_interest[!is.na(genes_of_interest)]

# Use first column (gene_id) for row names
counts <- data.frame(counts, row.names = 1)
counts$Ensembl_ID <- row.names(counts)
drop <- c("Ensembl_ID", "gene_name")
gene_info <- counts[, drop]
counts <- counts[, !(names(counts) %in% drop)] # remove both columns

# Import metadata
sample_metadata <- read_csv("data/MECFS_RNAseq_metadata_2023_08_17.csv")
row.names(sample_metadata) <- sample_metadata$RNA_Samples_id

# Check that data is ordered properly
sample_metadata <- check_order(sample_metadata = sample_metadata, counts = counts)

genes_biomart <- retrieve_gene_info(values = gene_info$Ensembl_ID, filters = "ensembl_gene_id_version")
```

## DESeq2 Analysis

```{r, start analysis}
sample_metadata$Family <- factor(sample_metadata$Family)
sample_metadata$Affected <- factor(sample_metadata$Affected)
sample_metadata$Batch <- factor(sample_metadata$Batch)
sample_metadata$Gender <- factor(sample_metadata$Gender)
sample_metadata$Tier <- factor(sample_metadata$Tier)
sample_metadata$Sex <- factor(sample_metadata$Sex)

# Create DESeqDataSet object using DESeqDataSetFromMatrix function
dds_tier <- DESeqDataSetFromMatrix(
  countData = round(counts),
  colData = sample_metadata,
  design = ~ Batch + Tier
)

# Pre-filtering: Keep only rows that have at least 10 reads total
keep <- rowSums(counts(dds_tier)) >= 10
dds_tier <- dds_tier[keep, ]

# Run DESeq function
dds_tier <- DESeq(dds_tier)

# Normalize gene counts for differences in sequencing depth/global differences
counts_norm_dds_tier <- counts(dds_tier, normalized = TRUE)

# Get the names of results
results_names <- resultsNames(dds_tier)

# Display normalized gene counts for the first few rows
head(counts_norm_dds_tier)
```

### Data transformation and visualization

Perform count data transformation by variance stabilizing transformation (vst) on normalized counts.

```{r, vsd}
vsd_tier <- vst(dds_tier, blind = FALSE)
```

### Batch correction with limma

```{r, limma}
counts_vst_tier <- assay(vsd_tier)
write.csv(counts_vst_tier, file = "output/counts_vst2.csv")
mm <- model.matrix(~ Batch + Tier, colData(vsd_tier))

counts_vst_tier_limma <- limma::removeBatchEffect(counts_vst_tier, batch = vsd_tier$Batch, design = mm)

write.csv(counts_vst_tier_limma, file = "output/counts_vst_limma.csv")

vsd_tier_limma <- vsd_tier
assay(vsd_tier_limma) <- counts_vst_tier_limma
```

### Comparison/Contrast of Tiers

```{r}
results_tier_1_vs_2 <- results(dds_tier, contrast = c("Tier", "1", "2"))
results_tier_1_vs_2 <- results_tier_1_vs_2[order(results_tier_1_vs_2$padj), ]
summary(results_tier_1_vs_2)
write.csv(results_tier_1_vs_2, file = "output/results_tier_1_vs_2.csv")
results_tier_1_vs_2_df <- as.data.frame(results_tier_1_vs_2)
results_tier_1_vs_2_df_05 <- subset(results_tier_1_vs_2_df, padj < 0.05)

results_tier_1_vs_3 <- results(dds_tier, contrast = c("Tier", "1", "3"))
results_tier_1_vs_3 <- results_tier_1_vs_3[order(results_tier_1_vs_3$padj), ]
summary(results_tier_1_vs_3)
write.csv(results_tier_1_vs_3, file = "output/results_tier_1_vs_3.csv")
results_tier_1_vs_3_df <- as.data.frame(results_tier_1_vs_3)
results_tier_1_vs_3_df_05 <- subset(results_tier_1_vs_3_df, padj < 0.05)
```

### Save data

```{r, save-data}
save.image(file = "output/tier-analysis.RData")
```
